
<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="utf-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<title>Watchdog's module library</title>
	<link rel="icon" href="gfx/watchdog.ico">
	<!-- CSS FILES -->
	<link rel="stylesheet" type="text/css" href="css/uikit.min.css">
	<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.10.19/css/jquery.dataTables.css">
	<link rel="stylesheet" type="text/css" href="css/modules.css">
</head>
<body>
	<div class="uk-offcanvas-content">
		<!--HEADER-->
		<header id="site-head">
			<div class="uk-container uk-container-expand">
				<div class="uk-grid uk-grid-medium uk-flex uk-flex-middle" data-uk-grid>
					<div class="uk-width-auto">
						<img class="uk-logo" src="gfx/watchdog.png" alt="" style="width: 64px" />
					</div>
					<div class="uk-width-expand">
						<div class="uk-inline uk-width-1.1">
							<span class="uk-form-icon uk-form-icon" data-uk-icon="icon: search"></span>
							<input class="uk-input uk-width-auto search-fld" type="search" placeholder="Search..." id="search-input" oninput="moduleFilter(true)">
							<span class="uk-width-auto"><input class="uk-checkbox" type="checkbox" id="search-name" checked onchange="moduleFilter(false)"/> title</span>
							<span class="uk-width-auto"><input class="uk-checkbox" type="checkbox" id="search-author" checked onchange="moduleFilter(false)"/> author</span>
							<span class="uk-width-auto"><input class="uk-checkbox" type="checkbox" id="search-category" checked onchange="moduleFilter(false)"/> category</span>
							<span class="uk-width-auto"><input class="uk-checkbox" type="checkbox" id="search-description" checked onchange="moduleFilter(false)"/> description</span>
						</div>
					</div>
					<div class="uk-width-auto">
						<ul class="uk-subnav" data-uk-margin>
							<li class="uk-visible@s"><a href="#">About</a></li>
						</ul>
					</div>
				</div>
			</div>
		</header>
		<!--/HEADER-->
<div class="spacer"></div>
<!--MODULE WRAPPER-->
<section id="filter-modules" class="uk-section uk-section-small uk-section-default uk-padding-remove-bottom" data-uk-filter="target: .js-filter">
	<div class="uk-container uk-container-expand uk-margin-large-bottom">
	<div class="uk-grid-small uk-grid-divider uk-child-width-auto uk-grid" uk-grid="">
	<div>
		<ul id="refresh-search" class="uk-subnav uk-subnavCategory uk-subnav-pill" uk-margin>
			<li class="uk-active" data-uk-filter-control="group: category"><a href="#" id="filter-category-link0">All Categories</a></li>
			<li data-uk-filter-control="filter: [meta-category*='§SEQUENCING§']; group: category"><a href="#" id="filter-category-link1">SEQUENCING (38)</a></li><!--FILTER_BUTTON_TEMPLATE_CATEGORY-->
			<li data-uk-filter-control="filter: [meta-category*='§GENERAL§']; group: category"><a href="#" id="filter-category-link2">GENERAL (9)</a></li><!--FILTER_BUTTON_TEMPLATE_CATEGORY-->
			<li data-uk-filter-control="filter: [meta-category*='§CHIP-SEQ§']; group: category"><a href="#" id="filter-category-link3">CHIP-SEQ (6)</a></li><!--FILTER_BUTTON_TEMPLATE_CATEGORY-->
			<li data-uk-filter-control="filter: [meta-category*='§CIRCRNA§']; group: category"><a href="#" id="filter-category-link4">CIRCRNA (5)</a></li><!--FILTER_BUTTON_TEMPLATE_CATEGORY-->
			<li data-uk-filter-control="filter: [meta-category*='§FILE UTILS§']; group: category"><a href="#" id="filter-category-link5">FILE UTILS (5)</a></li><!--FILTER_BUTTON_TEMPLATE_CATEGORY-->
		</ul>
	</div>
	<div>
		<ul class="uk-subnav uk-subnav-pill" uk-margin>
			<li class="uk-active" data-uk-filter-control="group: author"><a href="#" id="filter-author-link0">All Authors</a></li>
			<li data-uk-filter-control="filter: [meta-author*='§MICHAEL KLUGE§']; group: author"><a href="#" id="filter-author-link1">MICHAEL KLUGE (34)</a></li><!--FILTER_BUTTON_TEMPLATE_AUTHOR-->
			<li data-uk-filter-control="filter: [meta-author*='§SOPHIE FRIEDL§']; group: author"><a href="#" id="filter-author-link2">SOPHIE FRIEDL (20)</a></li><!--FILTER_BUTTON_TEMPLATE_AUTHOR-->
			<li data-uk-filter-control="filter: [meta-author*='§CAROLINE FRIEDEL§']; group: author"><a href="#" id="filter-author-link3">CAROLINE FRIEDEL (8)</a></li><!--FILTER_BUTTON_TEMPLATE_AUTHOR-->
			<li data-uk-filter-control="filter: [meta-author*='§DANIEL STROBL§']; group: author"><a href="#" id="filter-author-link4">DANIEL STROBL (1)</a></li><!--FILTER_BUTTON_TEMPLATE_AUTHOR-->
		</ul>
	</div>
	<div>
		<ul class="uk-subnav uk-subnav-pill" uk-margin>
			<li class="uk-active" data-uk-filter-control="group: age"><a href="#" id="filter-age-link0">All last update</a></li>
			<li data-uk-filter-control="filter: [meta-age*='§3 DAYS§']; group: age"><a href="#" id="filter-age-link1">3 DAYS (3)</a></li><!--FILTER_BUTTON_TEMPLATE_AGE-->
			<li data-uk-filter-control="filter: [meta-age*='§1 WEEK§']; group: age"><a href="#" id="filter-age-link2">1 WEEK (2)</a></li><!--FILTER_BUTTON_TEMPLATE_AGE-->
			<li data-uk-filter-control="filter: [meta-age*='§3 WEEKS§']; group: age"><a href="#" id="filter-age-link3">3 WEEKS (1)</a></li><!--FILTER_BUTTON_TEMPLATE_AGE-->
		</ul>
	</div>
	<!-- custom meta filter tag for text-search filter -->
	<span class="uk-active" data-uk-filter-control="filter: [meta-search=true]; group: search"></span>
</div>

	<div class="uk-grid uk-grid-medium uk-child-width-1-2@s uk-child-width-1-3@m uk-child-width-1-4@l  uk-child-width-1-6@xl uk-grid-match js-filter" data-uk-grid="masonry: true" data-uk-sortable="handle: .drag-icon">
		
<div class="cat2-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§GENERAL§" meta-update="2019-08-09" meta-search="true" meta-id="31">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">GENERAL</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-08-09
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full31" onclick="initFullModule(31, 1)" uk-toggle>AttachAnno</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">attaches annotations to a CSV file</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full31" class="uk-modal-full" meta-id="31" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>AttachAnno</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch31">attaches annotations to a CSV file</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU core utilities</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU R</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">packages: getopt, stringi</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable31" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>targetFile</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to char-separated table file with header</td><td>0</td><td>0</td></tr> 						<tr><td>targetSep</td><td>string</td><td></td><td>\t</td><td>1-</td><td>separating char in the annotation file(s)</td><td>0</td><td>0</td></tr> 						<tr><td>outputFile</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to the annotated output file</td><td>0</td><td>0</td></tr> 						<tr><td>targetIDcolumn</td><td>string</td><td></td><td></td><td>1-</td><td>name of the column of the target file that should be used to merge the table with the annotation file(s)</td><td>0</td><td>0</td></tr> 						<tr><td>annotationIDcolumn</td><td>string</td><td></td><td></td><td>1-</td><td>name(s) of the column(s) of the annotation file(s) that should be used to merge the table with the annotation file(s)</td><td>0</td><td>0</td></tr> 						<tr><td>annotationFile</td><td>file</td><td>absolute</td><td></td><td>1-</td><td>path(s) to annotation table file(s) that should be attached</td><td>0</td><td>0</td></tr> 						<tr><td>annotationSep</td><td>string</td><td></td><td>\t</td><td>1-</td><td>separating char in the target file</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable31" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§;§CHIP-SEQ§" meta-update="2019-11-14" meta-search="true" meta-id="42">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SEQUENCING</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-11-14
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full42" onclick="initFullModule(42, 1)" uk-toggle>ChIPSeeker</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">ChIPSeeker can be used to visualize called peaks in ChIP-seq data</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://www.bioconductor.org/packages/release/bioc/html/ChIPseeker.html" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/25765347" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full42" class="uk-modal-full" meta-id="42" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>ChIPSeeker</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch42">ChIPSeeker can be used to visualize called peaks in ChIP-seq data</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU R</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">packages: GenomicFeatures, ChIPseeker, getopt</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable42" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>bedFiles</td><td>file</td><td>absolute</td><td></td><td>1-</td><td>path to *.bed or *.narrowPeak files that contain called peaks</td><td>0</td><td>0</td></tr> 						<tr><td>annoDb</td><td>string</td><td></td><td></td><td>1</td><td>name of the R genome annotation database (e.g .org.Hs.eg.db)</td><td>0</td><td>0</td></tr> 						<tr><td>txdb</td><td>string</td><td></td><td></td><td>1</td><td>file or name of R library containing transcript-related features of a particular genome (e.g. TxDb.Hsapiens.UCSC.hg38.knownGene)</td><td>0</td><td>0</td></tr> 						<tr><td>outputDir</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to an output folder in which the plots will be stored</td><td>0</td><td>0</td></tr> 						<tr><td>promotorUpstream</td><td>integer</td><td></td><td>3000</td><td>*</td><td>size in bp used to define the promotor region upstream of the annotated TSS (transcription start site)</td><td>0</td><td>0</td></tr> 						<tr><td>promotorDownstream</td><td>integer</td><td></td><td>3000</td><td>*</td><td>size in bp used to define the promotor region downstream of the annotated TSS (transcription start site)</td><td>0</td><td>0</td></tr> 						<tr><td>resample</td><td>integer</td><td></td><td>1000</td><td>*</td><td>number of resample iterations for confidence interval estimation</td><td>0</td><td>0</td></tr> 						<tr><td>conf</td><td>string</td><td></td><td>0.95</td><td>*</td><td>confidence interval to be estimated</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable42" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>ChIPSeekerOutputFolder</td><td>string</td><td>path to the output folder containing the plots</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p></p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/25765347" data-uk-tooltip="title: PMID" target="_blank">25765347</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.bioconductor.org/packages/release/bioc/html/ChIPseeker.html" data-uk-tooltip="title: Website" target="_blank">https://www.bioconductor.org/packages/release/bioc/html/ChIPseeker.html</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2021-10-09" meta-search="true" meta-id="18">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SEQUENCING</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2021-10-09
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full18" onclick="initFullModule(18, 1)" uk-toggle>DETest</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">performs differential gene expression tests based on count tables</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="http://bioconductor.org/packages/release/bioc/html/DESeq.html" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/20979621" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full18" class="uk-modal-full" meta-id="18" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>DETest</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch18">performs differential gene expression tests based on count tables</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU R</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">packages: edgeR, DESeq, DESeq2, limma, Biobase, RColorBrewer, gplots, getopt, genefilter, lattice</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable18" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>controlCondition</td><td>string</td><td></td><td></td><td>1</td><td>name of the control condition</td><td>0</td><td>0</td></tr> 						<tr><td>testCondition</td><td>string</td><td></td><td></td><td>1</td><td>name of the test condition</td><td>0</td><td>0</td></tr> 						<tr><td>countFile</td><td>file</td><td>absolute</td><td></td><td>1</td><td>count file with features in rows and samples in columns</td><td>0</td><td>0</td></tr> 						<tr><td>sampleAnnotation</td><td>file</td><td>absolute</td><td></td><td>1</td><td>annotation file with sample names in the first colum and sample condition in the second condition; (header: sample\tcondition)</td><td>0</td><td>0</td></tr> 						<tr><td>featureAnnotation</td><td>file</td><td>absolute</td><td></td><td>*</td><td>annotation file which is joined with the count file</td><td>0</td><td>0</td></tr> 						<tr><td>featureAnnotationID</td><td>string</td><td></td><td>FeatureID</td><td>*</td><td>name of the column used for joining</td><td>0</td><td>0</td></tr> 						<tr><td>featureAnnotationType</td><td>string</td><td></td><td>type</td><td>*</td><td>name of the column in the annotation file for which a distribution plot is created</td><td>0</td><td>0</td></tr> 						<tr><td>excludeSamples</td><td>string</td><td></td><td></td><td>0-null</td><td>names of samples that should be excluded from the analysis</td><td>0</td><td>0</td></tr> 						<tr><td>pValueCutoff</td><td>double</td><td>[0-1]</td><td>0.01</td><td>*</td><td>p-Value cutoff for significant results</td><td>0</td><td>0</td></tr> 						<tr><td>minKeepReads</td><td>integer</td><td>&gt;=0</td><td>25</td><td>*</td><td>number of reads a feature must own in average per sample to pass filtering step before DE test is performed</td><td>0</td><td>0</td></tr> 						<tr><td>foldchangeCutoff</td><td>integer</td><td></td><td>0.0,0.415,1.0</td><td>0-null</td><td>log2 foldchange cutoffs for which a own result file will be created; will be used for both directions (+/-)</td><td>0</td><td>0</td></tr> 						<tr><td>foldchangeCutoffNames</td><td>string</td><td></td><td>significant,0.33-fold,2-fold</td><td>0-null</td><td>corresponding names to the foldchange cutoffs</td><td>0</td><td>0</td></tr> 						<tr><td>foldchangeCutoff</td><td>double</td><td></td><td>1</td><td>*</td><td>log2 foldchange cutoffs the two-colored volcano plot; will be used for both directions (+/-)</td><td>0</td><td>0</td></tr> 						<tr><td>downregColor</td><td>string</td><td></td><td>red</td><td>*</td><td>color for down-regulated genes in the two-colored volcano plot</td><td>0</td><td>0</td></tr> 						<tr><td>upregColor</td><td>string</td><td></td><td>blue</td><td>*</td><td>color for down-regulated genes in the two-colored volcano plot</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to output folder</td><td>0</td><td>0</td></tr> 						<tr><td>method</td><td>string</td><td></td><td>all</td><td>*</td><td>method that should be applied; one of: limma, DESeq, DESeq2, edgeR, all</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable18" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Differential gene expression analysis was performed using %method% (%SOFTWARE_VERSION%).</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/20979621" data-uk-tooltip="title: PMID" target="_blank">20979621</a>, 						<a href="https://www.ncbi.nlm.nih.gov/pubmed/19910308" data-uk-tooltip="title: PMID" target="_blank">19910308</a>, 						<a href="https://www.ncbi.nlm.nih.gov/pubmed/25605792" data-uk-tooltip="title: PMID" target="_blank">25605792</a>, 						<a href="https://www.ncbi.nlm.nih.gov/pubmed/25516281" data-uk-tooltip="title: PMID" target="_blank">25516281</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="http://bioconductor.org/packages/release/bioc/html/DESeq.html" data-uk-tooltip="title: Website" target="_blank">http://bioconductor.org/packages/release/bioc/html/DESeq.html</a><br /> 						<a href="http://bioconductor.org/packages/release/bioc/html/DESeq2.html" data-uk-tooltip="title: Website" target="_blank">http://bioconductor.org/packages/release/bioc/html/DESeq2.html</a><br /> 						<a href="http://bioconductor.org/packages/release/bioc/html/limma.html" data-uk-tooltip="title: Website" target="_blank">http://bioconductor.org/packages/release/bioc/html/limma.html</a><br /> 						<a href="http://bioconductor.org/packages/release/bioc/html/DESeq.html" data-uk-tooltip="title: Website" target="_blank">http://bioconductor.org/packages/release/bioc/html/DESeq.html</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-08-08" meta-search="true" meta-id="27">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SEQUENCING</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-08-08
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full27" onclick="initFullModule(27, 1)" uk-toggle>DEXSeq</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">tests RNA-seq data for differential exon usage</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://bioconductor.org/packages/release/bioc/html/DEXSeq.html" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/22722343" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full27" class="uk-modal-full" meta-id="27" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>DEXSeq</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch27">tests RNA-seq data for differential exon usage</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU core utils</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU R</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">packages: getopt, DEXSeq, GenomicFeatures, BiocParallel, GenomicRanges, GenomicFeatures</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable27" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>controlCondition</td><td>string</td><td></td><td></td><td>1</td><td>name of the control condition</td><td>0</td><td>0</td></tr> 						<tr><td>testCondition</td><td>string</td><td></td><td></td><td>1</td><td>name of the test condition</td><td>0</td><td>0</td></tr> 						<tr><td>countFile</td><td>file</td><td>absolute</td><td></td><td>1</td><td>count file with features in rows and samples in columns</td><td>0</td><td>0</td></tr> 						<tr><td>flattedGTFAnnotation</td><td>file</td><td>absolute</td><td></td><td>1</td><td>flatted GTF file which was used to create the count file; created by dexseq_prepare_annotation.py that comes with DEXSeq</td><td>0</td><td>0</td></tr> 						<tr><td>sampleAnnotation</td><td>file</td><td>absolute</td><td></td><td>1</td><td>annotation file with sample names in the first colum and sample condition in the second condition (header: sample\tcondition)</td><td>0</td><td>0</td></tr> 						<tr><td>featureAnnotation</td><td>file</td><td>absolute</td><td></td><td>*</td><td>annotation file which is joined with the count file</td><td>0</td><td>0</td></tr> 						<tr><td>featureAnnotationID</td><td>string</td><td></td><td>Geneid</td><td>*</td><td>name of the column with is used for joining</td><td>0</td><td>0</td></tr> 						<tr><td>featureAnnotationName</td><td>string</td><td></td><td>name</td><td>*</td><td>name of the column in the annotation file that contains the name of the feature</td><td>0</td><td>0</td></tr> 						<tr><td>excludeSamples</td><td>string</td><td></td><td></td><td>0-null</td><td>names of samples that should be excluded from the analysis</td><td>0</td><td>0</td></tr> 						<tr><td>pValueCutoff</td><td>double</td><td>[0,1]</td><td>0.01</td><td>*</td><td>p-Value cutoff for significant results</td><td>0</td><td>0</td></tr> 						<tr><td>minKeepReads</td><td>integer</td><td>[1,]</td><td>25</td><td>*</td><td>number of reads a feature must own in average per sample to pass filtering step before DE test is performed</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>file</td><td>absolute</td><td></td><td>1</td><td>output folder</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td>[1,]</td><td>1</td><td>*</td><td>number of threads to use for testing</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable27" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Differential exon usage was determined using DEXSeq (%SOFTWARE_VERSION%).</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/22722343" data-uk-tooltip="title: PMID" target="_blank">22722343</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://bioconductor.org/packages/release/bioc/html/DEXSeq.html" data-uk-tooltip="title: Website" target="_blank">https://bioconductor.org/packages/release/bioc/html/DEXSeq.html</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-08-08" meta-search="true" meta-id="47">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SEQUENCING</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-08-08
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full47" onclick="initFullModule(47, 1)" uk-toggle>DaPars</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">dynamic analysis of alternative polyadenylation from RNA-seq</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://github.com/ZhengXia/dapars" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/25409906" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full47" class="uk-modal-full" meta-id="47" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>DaPars</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch47">dynamic analysis of alternative polyadenylation from RNA-seq</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">DaPars</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU core utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable47" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>controlCondition</td><td>string</td><td></td><td></td><td>1</td><td>name of the control condition</td><td>0</td><td>0</td></tr> 						<tr><td>testCondition</td><td>string</td><td></td><td></td><td>1</td><td>name of the test condition</td><td>0</td><td>0</td></tr> 						<tr><td>sampleAnnotation</td><td>file</td><td>absolute</td><td></td><td>1</td><td>annotation file with sample names in the first colum and sample condition in the second condition (header: sample\tcondition)</td><td>0</td><td>0</td></tr> 						<tr><td>excludeSamples</td><td>string</td><td></td><td></td><td>1-</td><td>names of samples that should be excluded from the analysis</td><td>0</td><td>0</td></tr> 						<tr><td>wigFolder</td><td>file</td><td></td><td></td><td>1</td><td>folder containing the wig files (format: folder/samplename.bedgraph)</td><td>0</td><td>0</td></tr> 						<tr><td>wigEnding</td><td>string</td><td></td><td>bedgraph</td><td>*</td><td>ending of the wig files</td><td>0</td><td>0</td></tr> 						<tr><td>annotated3UTR</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to annotated 3' UTR regions created with DaPars_Extract_Anno.py</td><td>0</td><td>0</td></tr> 						<tr><td>outputFile</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to the output file</td><td>0</td><td>0</td></tr> 						<tr><td>coverageCutoff</td><td>integer</td><td>[1,]</td><td>30</td><td>*</td><td>coverage threshold</td><td>0</td><td>0</td></tr> 						<tr><td>FDRCutoff</td><td>double</td><td>[0,1]</td><td>0.01</td><td>*</td><td>FDR cutoff</td><td>0</td><td>0</td></tr> 						<tr><td>PDUICutoff</td><td>double</td><td>[0,100]</td><td>0.5</td><td>*</td><td>degree of difference in APA usage in percent</td><td>0</td><td>0</td></tr> 						<tr><td>FoldChangeCutoff</td><td>double</td><td></td><td>0.5</td><td>*</td><td>log2 foldchange cutoff between the two conditions</td><td>0</td><td>0</td></tr> 						<tr><td>numberOfCondASamplesReachingCutoff</td><td>integer</td><td>[1,]</td><td></td><td>*</td><td>number of samples from condition A that must pass the coverage cutoff; default: all samples</td><td>0</td><td>0</td></tr> 						<tr><td>numberOfCondBSamplesReachingCutoff</td><td>integer</td><td>[1,]</td><td></td><td>*</td><td>number of samples from condition B that must pass the coverage cutoff; default: all samples</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable47" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>wiggleFile</td><td>string</td><td>path to the output file in WIG format</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>DaPars (%SOFTWARE_VERSION%) was used to identify alternative polyadenylation.</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/25409906" data-uk-tooltip="title: PMID" target="_blank">25409906</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://github.com/ZhengXia/dapars" data-uk-tooltip="title: Website" target="_blank">https://github.com/ZhengXia/dapars</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-08-08" meta-search="true" meta-id="46">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SEQUENCING</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-08-08
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full46" onclick="initFullModule(46, 1)" uk-toggle>EnrichAnno</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">gene set enrichment analysis on GO and KEGG</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://github.com/GuangchuangYu/clusterProfiler" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/22455463" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full46" class="uk-modal-full" meta-id="46" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>EnrichAnno</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch46">gene set enrichment analysis on GO and KEGG</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU R</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">packages: getopt, clusterProfiler, pathview, KEGGREST</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable46" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>backgroundFile</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to file with header, which contains a list of ENSEMBL or GENDCODE identifiers that should be used as backgroud</td><td>0</td><td>0</td></tr> 						<tr><td>testFiles</td><td>string</td><td>absolute</td><td></td><td>1-</td><td>path to file(s) with header, which contain a list of ENSEMBL or GENDCODE identifiers that should be used for enrichment testing</td><td>0</td><td>0</td></tr> 						<tr><td>orgDB</td><td>string</td><td></td><td></td><td>1</td><td>name of the organism database (orgDB) that should be used as GO annotation; if package is missing it is installed via biocLite</td><td>0</td><td>0</td></tr> 						<tr><td>keggDBName</td><td>string</td><td></td><td></td><td>*</td><td>organism code for KEGG (e.g. mmu / hsa); http://www.genome.jp/kegg/catalog/org_list.html; if not supported by KEGGREST parameter will be ignored</td><td>0</td><td>0</td></tr> 						<tr><td>pValueCutoff</td><td>double</td><td></td><td>0.01</td><td>*</td><td>p-Value cutoff for significant results</td><td>0</td><td>0</td></tr> 						<tr><td>plotKegg</td><td>boolean</td><td></td><td>true</td><td>*</td><td>if enabled, plots are created for KEGG pathways</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to output basename; folder is created if not existent</td><td>0</td><td>0</td></tr> 						<tr><td>suffix</td><td>string</td><td></td><td></td><td>*</td><td>suffix that is inserted before basename of output; if a absolute path basename is applied</td><td>0</td><td>0</td></tr> 						<tr><td>foldchangeCol</td><td>string</td><td></td><td></td><td>*</td><td>name of the colum that contains the log2FC</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable46" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Afterwards gene set enrichment analysis was performed on gene sets defined by GO (%orgDB%) and KEGG (%keggDBName%) enrichment on up-/down-regulated genes using clusterProfiler (%SOFTWARE_VERSION%).</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/22455463" data-uk-tooltip="title: PMID" target="_blank">22455463</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://github.com/GuangchuangYu/clusterProfiler" data-uk-tooltip="title: Website" target="_blank">https://github.com/GuangchuangYu/clusterProfiler</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§;§CHIP-SEQ§" meta-update="2019-11-08" meta-search="true" meta-id="15">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SEQUENCING</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-11-08
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full15" onclick="initFullModule(15, 1)" uk-toggle>GEM</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">identifies protein-DNA interaction at high resolution in ChIP-seq data</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://groups.csail.mit.edu/cgs/gem/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/22912568" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full15" class="uk-modal-full" meta-id="15" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>GEM</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch15">identifies protein-DNA interaction at high resolution in ChIP-seq data</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GEM</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">java</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU core utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable15" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>jarPath</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to GEM jar file</td><td>0</td><td>0</td></tr> 						<tr><td>expt</td><td>file</td><td>absolute</td><td></td><td>1</td><td>aligned read file</td><td>0</td><td>0</td></tr> 						<tr><td>readDistribution</td><td>file</td><td>absolute</td><td></td><td>1</td><td>read spatial distribution file</td><td>0</td><td>0</td></tr> 						<tr><td>gpsOnly</td><td>boolean</td><td></td><td>true</td><td>*</td><td>run in GPS only mode</td><td>0</td><td>0</td></tr> 						<tr><td>k</td><td>integer</td><td></td><td>8</td><td>*</td><td>length of the k-mer for motif finding, use --k or (--kmin &amp; --kmax); GEM parameter</td><td>0</td><td>0</td></tr> 						<tr><td>kMin</td><td>integer</td><td></td><td>6</td><td>*</td><td>min value of k, e.g. 6; GEM parameter</td><td>0</td><td>0</td></tr> 						<tr><td>kMax</td><td>integer</td><td></td><td>13</td><td>*</td><td>max value of k, e.g. 13; GEM parameter</td><td>0</td><td>0</td></tr> 						<tr><td>seed</td><td>string</td><td></td><td></td><td>*</td><td>exact k-mer string to jump start k-mer set motif discovery; GEM parameter</td><td>0</td><td>0</td></tr> 						<tr><td>genome</td><td>folder</td><td>absolute</td><td></td><td>*</td><td>the path to the genome sequence directory, for motif finding; GEM parameter</td><td>0</td><td>0</td></tr> 						<tr><td>outputPrefix</td><td>file</td><td>absolute</td><td></td><td>1</td><td>output folder name and file name prefix</td><td>0</td><td>0</td></tr> 						<tr><td>control</td><td>file</td><td>absolute</td><td></td><td>*</td><td>aligned reads file for control</td><td>0</td><td>0</td></tr> 						<tr><td>chrSize</td><td>file</td><td>absolute</td><td></td><td>*</td><td>genome chrom.sizes file with chr name/length pairs</td><td>0</td><td>0</td></tr> 						<tr><td>format</td><td>string</td><td></td><td>BED</td><td>*</td><td>read file format: BED/SAM/BOWTIE/ELAND/NOVO</td><td>0</td><td>0</td></tr> 						<tr><td>sizeInBp</td><td>integer</td><td></td><td> </td><td>*</td><td>size of mappable genome in bp (default is estimated from genome chrom sizes)</td><td>0</td><td>0</td></tr> 						<tr><td>alphaValue</td><td>double</td><td></td><td> </td><td>*</td><td>minimum alpha value for sparse prior (default is esitmated from the whole dataset coverage)</td><td>0</td><td>0</td></tr> 						<tr><td>qValue</td><td>double</td><td></td><td>2</td><td>*</td><td>significance level for q-value, specify as -log10(q-value) (default=2, q-value=0.01)</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td></td><td>#CPU</td><td>*</td><td>maximum number of threads to run GEM in paralell</td><td>0</td><td>0</td></tr> 						<tr><td>kSeqs</td><td>integer</td><td></td><td>5000</td><td>*</td><td>number of binding events to use for motif discovery; GEM parameter</td><td>0</td><td>0</td></tr> 						<tr><td>memoryPerThread</td><td>integer</td><td></td><td>2048</td><td>*</td><td>total memory per thread in MB if running on local host; otherwise memory limit of executor might be set</td><td>0</td><td>0</td></tr> 						<tr><td>useFixedAlpha</td><td>boolean</td><td></td><td>false</td><td>*</td><td>use a fixed user-specified alpha value for all the regions</td><td>0</td><td>0</td></tr> 						<tr><td>JASPAROutput</td><td>boolean</td><td></td><td>true</td><td>*</td><td>output motif PFM in JASPAR format; GEM parameter</td><td>0</td><td>0</td></tr> 						<tr><td>MEMEOutput</td><td>boolean</td><td></td><td>true</td><td>*</td><td>output motif PFM in MEME format; GEM parameter</td><td>0</td><td>0</td></tr> 						<tr><td>HOMEROutput</td><td>boolean</td><td></td><td>true</td><td>*</td><td>output motif PFM in HOMER format; GEM parameter</td><td>0</td><td>0</td></tr> 						<tr><td>BEDOutput</td><td>boolean</td><td></td><td>true</td><td>*</td><td>output binding events in BED format for UCSC Genome Browser</td><td>0</td><td>0</td></tr> 						<tr><td>NarrowPeakOutput</td><td>boolean</td><td></td><td>true</td><td>*</td><td>output binding events in ENCODE NarrowPeak format </td><td>0</td><td>0</td></tr> 						<tr><td>workingDir</td><td>folder path</td><td>absolute</td><td>/usr/local/storage/</td><td>*</td><td>path to working directory</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable15" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>GEM (%SOFTWARE_VERSION%) was used to call peaks in the ChIP-seq data [Y. Guo, S. Mahony, D.K. Gifford, High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. PLoS Computational Biology, (2012) 8(8): e1002638].</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/22912568" data-uk-tooltip="title: PMID" target="_blank">22912568</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://groups.csail.mit.edu/cgs/gem/" data-uk-tooltip="title: Website" target="_blank">https://groups.csail.mit.edu/cgs/gem/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§DANIEL STROBL§;§CAROLINE FRIEDEL§" meta-category="§SEQUENCING§" meta-update="2019-10-16" meta-search="true" meta-id="25">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-10-16
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full25" onclick="initFullModule(25, 1)" uk-toggle>HISAT2</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">Performs spliced RNA-seq read mapping using HISAT2.</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Daniel Strobl</div>
				<div class="uk-width-auto">
					<a href="WEBSITE [1-]" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/25751142" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full25" class="uk-modal-full" meta-id="25" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>HISAT2</h1>
					<h3><i>by</i> Daniel Strobl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch25">Performs spliced RNA-seq read mapping using HISAT2.</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">HISAT2</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable25" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>unpaired</td><td>file</td><td></td><td></td><td>*</td><td>Files with unpaired reads. Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).</td><td>0</td><td>0</td></tr> 						<tr><td>s</td><td>integer</td><td></td><td></td><td>*</td><td>skip the first &lt;int&gt; reads/pairs in the input (none)</td><td>0</td><td>0</td></tr> 						<tr><td>u</td><td>integer</td><td></td><td></td><td>*</td><td>stop after first &lt;int&gt; reads/pairs (no limit)</td><td>0</td><td>0</td></tr> 						<tr><td>trim5</td><td>string</td><td></td><td></td><td>*</td><td>trim &lt;int&gt; bases from 5'/left end of reads (0)</td><td>0</td><td>0</td></tr> 						<tr><td>trim3</td><td>string</td><td></td><td></td><td>*</td><td>trim &lt;int&gt; bases from 3'/right end of reads (0)</td><td>0</td><td>0</td></tr> 						<tr><td>nceil</td><td>string</td><td></td><td></td><td>*</td><td>func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)</td><td>0</td><td>0</td></tr> 						<tr><td>pencansplice</td><td>integer</td><td></td><td></td><td>*</td><td>penalty for a canonical splice site (0)</td><td>0</td><td>0</td></tr> 						<tr><td>pennoncansplice</td><td>integer</td><td></td><td></td><td>*</td><td>penalty for a non-canonical splice site (12)</td><td>0</td><td>0</td></tr> 						<tr><td>pencanintronlen</td><td>string</td><td></td><td></td><td>*</td><td>penalty for long introns (G,-8,1) with canonical splice sites</td><td>0</td><td>0</td></tr> 						<tr><td>pennoncanintronlen</td><td>string</td><td></td><td></td><td>*</td><td>penalty for long introns (G,-8,1) with noncanonical splice sites</td><td>0</td><td>0</td></tr> 						<tr><td>minintronlen</td><td>integer</td><td></td><td></td><td>*</td><td>minimum intron length (20)</td><td>0</td><td>0</td></tr> 						<tr><td>maxintronlen</td><td>integer</td><td></td><td></td><td>*</td><td>maximum intron length (500000)</td><td>0</td><td>0</td></tr> 						<tr><td>knownsplicesiteinfile</td><td>file</td><td></td><td></td><td>*</td><td>provide a list of known splice sites</td><td>0</td><td>0</td></tr> 						<tr><td>novelsplicesiteoutfile</td><td>file</td><td></td><td></td><td>*</td><td>report a list of splice sites</td><td>0</td><td>0</td></tr> 						<tr><td>rnastrandness</td><td>string</td><td></td><td></td><td>*</td><td>Specify strand-specific information (unstranded)</td><td>0</td><td>0</td></tr> 						<tr><td>ma</td><td>integer</td><td></td><td></td><td>*</td><td>match bonus (0 for --end-to-end, 2 for --local)</td><td>0</td><td>0</td></tr> 						<tr><td>mp</td><td>string</td><td></td><td></td><td>*</td><td>max and min penalties for mismatch; lower qual = lower penalty &lt;6,2&gt;</td><td>0</td><td>0</td></tr> 						<tr><td>sp</td><td>string</td><td></td><td></td><td>*</td><td>max and min penalties for soft-clipping; lower qual = lower penalty &lt;2,1&gt;</td><td>0</td><td>0</td></tr> 						<tr><td>np</td><td>integer</td><td></td><td></td><td>*</td><td>penalty for non-A/C/G/Ts in read/ref (1)</td><td>0</td><td>0</td></tr> 						<tr><td>rdg</td><td>string</td><td></td><td></td><td>*</td><td>read gap open, extend penalties (5,3)</td><td>0</td><td>0</td></tr> 						<tr><td>rfg</td><td>string</td><td></td><td></td><td>*</td><td>reference gap open, extend penalties (5,3)</td><td>0</td><td>0</td></tr> 						<tr><td>scoremin</td><td>string</td><td></td><td></td><td>*</td><td>min acceptable alignment score w/r/t read length (L,0.0,-0.2)</td><td>0</td><td>0</td></tr> 						<tr><td>k</td><td>integer</td><td></td><td></td><td>*</td><td>report up to &lt;int&gt; alns per read; MAPQ not meaningful</td><td>0</td><td>0</td></tr> 						<tr><td>a</td><td>integer</td><td></td><td></td><td>*</td><td>report all alignments; very slow, MAPQ not meaningful</td><td>0</td><td>0</td></tr> 						<tr><td>un</td><td>file</td><td></td><td></td><td>*</td><td>write unpaired reads that didn't align to &lt;path&gt;</td><td>0</td><td>0</td></tr> 						<tr><td>al</td><td>file</td><td></td><td></td><td>*</td><td>write unpaired reads that aligned at least once to &lt;path&gt;</td><td>0</td><td>0</td></tr> 						<tr><td>unconc</td><td>file</td><td></td><td></td><td>*</td><td>write pairs that didn't align concordantly to &lt;path&gt;</td><td>0</td><td>0</td></tr> 						<tr><td>alconc</td><td>file</td><td></td><td></td><td>*</td><td>write pairs that aligned concordantly at least once to &lt;path&gt;</td><td>0</td><td>0</td></tr> 						<tr><td>metfile</td><td>file</td><td></td><td></td><td>*</td><td>send metrics to file at &lt;path&gt; (off)</td><td>0</td><td>0</td></tr> 						<tr><td>met</td><td>integer</td><td></td><td></td><td>*</td><td>report internal counters &amp; metrics every &lt;int&gt; secs (1)</td><td>0</td><td>0</td></tr> 						<tr><td>rgid</td><td>string</td><td></td><td></td><td>*</td><td>set read group id, reflected in @RG line and RG:Z: opt field</td><td>0</td><td>0</td></tr> 						<tr><td>rg</td><td>string</td><td></td><td></td><td>*</td><td>add &lt;text&gt; (\&quot;lab:value\&quot;) to @RG line of SAM header.</td><td>0</td><td>0</td></tr> 						<tr><td>offrate</td><td>integer</td><td></td><td></td><td>*</td><td>override offrate of index; must be &gt;= index's offrate</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td></td><td></td><td>*</td><td>number of alignment threads to launch (1)</td><td>0</td><td>0</td></tr> 						<tr><td>seed</td><td>integer</td><td></td><td></td><td>*</td><td>seed for random number generator (0)</td><td>0</td><td>0</td></tr> 						<tr><td>index</td><td>string</td><td></td><td></td><td>1</td><td>Index filename prefix (minus trailing .X.ht2)</td><td>0</td><td>0</td></tr> 						<tr><td>paired1</td><td>file</td><td></td><td></td><td>*</td><td>Files with #1 mates, paired with files in &lt;m2&gt;. Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).</td><td>0</td><td>0</td></tr> 						<tr><td>paired2</td><td>file</td><td></td><td></td><td>*</td><td>Files with #2 mates, paired with files in &lt;m1&gt;. Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>file</td><td></td><td></td><td>1</td><td>File for SAM output</td><td>0</td><td>0</td></tr> 						<tr><td>fastq</td><td>boolean</td><td></td><td>false</td><td>*</td><td>query input files are FASTQ .fq/.fastq (default)</td><td>0</td><td>0</td></tr> 						<tr><td>qseq</td><td>boolean</td><td></td><td>false</td><td>*</td><td>query input files are in Illumina's qseq format</td><td>0</td><td>0</td></tr> 						<tr><td>fasta</td><td>boolean</td><td></td><td>false</td><td>*</td><td>query input files are (multi-)FASTA .fa/.mfa</td><td>0</td><td>0</td></tr> 						<tr><td>raw</td><td>boolean</td><td></td><td>false</td><td>*</td><td>query input files are raw one-sequence-per-line</td><td>0</td><td>0</td></tr> 						<tr><td>c</td><td>boolean</td><td></td><td>false</td><td>*</td><td>paired1, paired2, unpaired are sequences themselves, not files</td><td>0</td><td>0</td></tr> 						<tr><td>phred33</td><td>boolean</td><td></td><td>false</td><td>*</td><td>qualities are Phred+33 (default)</td><td>0</td><td>0</td></tr> 						<tr><td>phred64</td><td>boolean</td><td></td><td>false</td><td>*</td><td>qualities are Phred+64</td><td>0</td><td>0</td></tr> 						<tr><td>intquals</td><td>boolean</td><td></td><td>false</td><td>*</td><td>qualities encoded as space-delimited integers</td><td>0</td><td>0</td></tr> 						<tr><td>ignorequals</td><td>boolean</td><td></td><td>false</td><td>*</td><td>treat all quality values as 30 on Phred scale (off)</td><td>0</td><td>0</td></tr> 						<tr><td>nofw</td><td>boolean</td><td></td><td>false</td><td>*</td><td>do not align forward (original) version of read (off)</td><td>0</td><td>0</td></tr> 						<tr><td>norc</td><td>boolean</td><td></td><td>false</td><td>*</td><td>do not align reverse-complement version of read (off)</td><td>0</td><td>0</td></tr> 						<tr><td>novelsplicesiteinfile</td><td>boolean</td><td></td><td>false</td><td>*</td><td>provide a list of novel splice sites</td><td>0</td><td>0</td></tr> 						<tr><td>notempsplicesite</td><td>boolean</td><td></td><td>false</td><td>*</td><td>disable the use of splice sites found</td><td>0</td><td>0</td></tr> 						<tr><td>nosplicedalignment</td><td>boolean</td><td></td><td>false</td><td>*</td><td>disable spliced alignment</td><td>0</td><td>0</td></tr> 						<tr><td>tmo</td><td>boolean</td><td></td><td>false</td><td>*</td><td>Reports only those alignments within known transcriptome</td><td>0</td><td>0</td></tr> 						<tr><td>dta</td><td>boolean</td><td></td><td>false</td><td>*</td><td>Reports alignments tailored for transcript assemblers</td><td>0</td><td>0</td></tr> 						<tr><td>dtacufflinks</td><td>boolean</td><td></td><td>false</td><td>*</td><td>Reports alignments tailored specifically for cufflinks</td><td>0</td><td>0</td></tr> 						<tr><td>fr</td><td>boolean</td><td></td><td>false</td><td>*</td><td>-1, -2 mates align fw/rev</td><td>0</td><td>0</td></tr> 						<tr><td>nomixed</td><td>boolean</td><td></td><td>false</td><td>*</td><td>suppress unpaired alignments for paired reads</td><td>0</td><td>0</td></tr> 						<tr><td>nodiscordant</td><td>boolean</td><td></td><td>false</td><td>*</td><td>suppress discordant alignments for paired reads</td><td>0</td><td>0</td></tr> 						<tr><td>t</td><td>boolean</td><td></td><td>false</td><td>*</td><td>print wall-clock time taken by search phases</td><td>0</td><td>0</td></tr> 						<tr><td>quiet</td><td>boolean</td><td></td><td>false</td><td>*</td><td>print nothing to stderr except serious errors</td><td>0</td><td>0</td></tr> 						<tr><td>metstderr</td><td>boolean</td><td></td><td>false</td><td>*</td><td>send metrics to stderr (off)</td><td>0</td><td>0</td></tr> 						<tr><td>nohead</td><td>boolean</td><td></td><td>false</td><td>*</td><td>supppress header lines, i.e. lines starting with @</td><td>0</td><td>0</td></tr> 						<tr><td>nosq</td><td>boolean</td><td></td><td>false</td><td>*</td><td>supppress @SQ header lines</td><td>0</td><td>0</td></tr> 						<tr><td>omitsecseq</td><td>boolean</td><td></td><td>false</td><td>*</td><td>put '*' in SEQ and QUAL fields for secondary alignments.</td><td>0</td><td>0</td></tr> 						<tr><td>reorder</td><td>boolean</td><td></td><td>false</td><td>*</td><td>force SAM output order to match order of input reads</td><td>0</td><td>0</td></tr> 						<tr><td>mm</td><td>boolean</td><td></td><td>false</td><td>*</td><td>use memory-mapped I/O for index; many 'bowtie's can share</td><td>0</td><td>0</td></tr> 						<tr><td>qcfilter</td><td>boolean</td><td></td><td>false</td><td>*</td><td>filter out reads that are bad according to QSEQ filter</td><td>0</td><td>0</td></tr> 						<tr><td>nondeterministic</td><td>boolean</td><td></td><td>false</td><td>*</td><td>seed rand. gen. arbitrarily instead of using read attributes</td><td>0</td><td>0</td></tr> 						<tr><td>removechrname</td><td>boolean</td><td></td><td>false</td><td>*</td><td>remove 'chr' from reference names in alignment</td><td>0</td><td>0</td></tr> 						<tr><td>addchrname</td><td>boolean</td><td></td><td>false</td><td>*</td><td>add 'chr' to reference names in alignment</td><td>0</td><td>0</td></tr> 						<tr><td>rf</td><td>boolean</td><td></td><td>false</td><td>*</td><td>-1, -2 mates align rev/fw</td><td>0</td><td>0</td></tr> 						<tr><td>ff</td><td>boolean</td><td></td><td>false</td><td>*</td><td>-1, -2 mates align fw/fw</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable25" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>SAMFile</td><td>string</td><td>output SAM file (= value for parameter output)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Sequencing reads were mapped using HISAT2 (version (%SOFTWARE_VERSION%)) [Kim D, Langmead B, Salzberg SL. HISAT: a fast spliced aligner with low memory 
requirements. Nat Methods. 2015 Apr;12(4):357-60].</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/25751142" data-uk-tooltip="title: PMID" target="_blank">25751142</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="WEBSITE [1-]" data-uk-tooltip="title: Website" target="_blank">WEBSITE [1-]</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§CAROLINE FRIEDEL§" meta-category="§SEQUENCING§" meta-update="2022-03-23" meta-search="true" meta-id="6">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2022-03-23
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full6" onclick="initFullModule(6, 1)" uk-toggle>STARgenomeGenerate</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">Generation of genome indices for STAR</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Caroline Friedel</div>
				<div class="uk-width-auto">
					<a href="https://github.com/alexdobin/STAR" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/23104886" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full6" class="uk-modal-full" meta-id="6" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>STARgenomeGenerate</h1>
					<h3><i>by</i> Caroline Friedel - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch6">Generation of genome indices for STAR</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">STAR</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable6" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>runThreadN</td><td>integer</td><td></td><td></td><td>*</td><td>[optional] int: number of threads to run STAR</td><td>0</td><td>0</td></tr> 						<tr><td>genomeDir</td><td>file</td><td></td><td></td><td>1</td><td>string: path to the directory where genome files will be generated</td><td>0</td><td>0</td></tr> 						<tr><td>genomeFastaFiles</td><td>file</td><td></td><td></td><td>1-</td><td>string(s): path(s) to the fasta files with the genome sequences, separated by spaces. These files should be plain text FASTA files, they *cannot* be zipped.</td><td>0</td><td>0</td></tr> 						<tr><td>sjdbGTFfile</td><td>file</td><td></td><td></td><td>*</td><td>[optional] string: path to the GTF file with annotations</td><td>0</td><td>0</td></tr> 						<tr><td>sjdbOverhang</td><td>integer</td><td></td><td>100</td><td>*</td><td>[optional] int&amp;amp;gt;0: length of the donor/acceptor sequence on each side of the junctions, ideally = (mate_length - 1)</td><td>0</td><td>0</td></tr> 						<tr><td>sjdbGTFtagExonParentTranscript</td><td>string</td><td></td><td>transcript_id</td><td>*</td><td>[optional] string: GTF attribute name for parent transcript ID (default &amp;amp;quot;transcript_id&amp;amp;quot; works for GTF files)</td><td>0</td><td>0</td></tr> 						<tr><td>sjdbFileChrStartEnd</td><td>file</td><td></td><td></td><td>0-null</td><td>[optional] string(s): path to the files with genomic coordinates (chr &amp;amp;lt;tab&amp;amp;gt; start &amp;amp;lt;tab&amp;amp;gt; end &amp;amp;lt;tab&amp;amp;gt; strand) for the splice junction introns.</td><td>0</td><td>0</td></tr> 						<tr><td>genomeSAindexNbases</td><td>integer</td><td></td><td></td><td>*</td><td>[optional] int: length (bases) of the SA pre-indexing string. Typically between 10 and 15. Longer strings will use much more memory, but allow faster searches. For small genomes, the parameter --genomeSAindexNbases must be scaled down to min(14, log2(GenomeLength)/2 - 1).</td><td>0</td><td>0</td></tr> 						<tr><td>genomeChrBinNbits</td><td>integer</td><td></td><td></td><td>*</td><td>[optional] int: =log2(chrBin), where chrBin is the size of the bins for genome storage: each chromosome will occupy an integer number of bins. For a genome with large number of contigs, it is recommended to scale this parameter as min(18, log2[max(GenomeLength/NumberOfReferences,ReadLength)]).</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable6" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>STAR indices were created for the  XXX genom using STAR (%SOFTWARE_VERSION%).</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/23104886" data-uk-tooltip="title: PMID" target="_blank">23104886</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://github.com/alexdobin/STAR" data-uk-tooltip="title: Website" target="_blank">https://github.com/alexdobin/STAR</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-03-13" meta-search="true" meta-id="56">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-03-13
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full56" onclick="initFullModule(56, 1)" uk-toggle>addSequence2Sam</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">sequences (and qualities) of FASTQ files can be added to SAM files</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full56" class="uk-modal-full" meta-id="56" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>addSequence2Sam</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch56">sequences (and qualities) of FASTQ files can be added to SAM files</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">perl</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable56" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>sam</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to the SAM file</td><td>0</td><td>0</td></tr> 						<tr><td>fastq</td><td>file path</td><td>absolute</td><td></td><td>1-</td><td>path to the FASTQ file(s)</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to the output SAM file in which the sequences are added</td><td>0</td><td>0</td></tr> 						<tr><td>unmapped</td><td>file path</td><td>absolute</td><td></td><td>*</td><td>path to a FASTQ file in which the unmapped sequences will be written to; exclusive with --preread flag</td><td>0</td><td>0</td></tr> 						<tr><td>noquality</td><td>boolean</td><td></td><td>false</td><td>*</td><td>does not add the read quality values</td><td>0</td><td>0</td></tr> 						<tr><td>update</td><td>boolean</td><td></td><td>false</td><td>*</td><td>overrides already existing output files</td><td>0</td><td>0</td></tr> 						<tr><td>preread</td><td>boolean</td><td></td><td>false</td><td>*</td><td>does only index reads stored in the FASTQ file that are part of the SAM file; exclusive with --unmapped parameter</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable56" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>SAMFileWithSequences</td><td>string</td><td>absolute path to the SAM file with added sequences and, if enabled, qualities</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>UnmappedReadFile</td><td>string</td><td>absolute path to file containing all unmapped reads in FASTQ format</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-08-08" meta-search="true" meta-id="60">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SEQUENCING</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-08-08
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full60" onclick="initFullModule(60, 1)" uk-toggle>bam2wiggle</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">converts BAM files to WIG files</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://bedtools.readthedocs.io/en/latest" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/20110278" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full60" class="uk-modal-full" meta-id="60" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>bam2wiggle</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch60">converts BAM files to WIG files</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">bedtools</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU core utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable60" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>bam</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to the position-based-sorted BAM file</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to BEDGRAPH file</td><td>0</td><td>0</td></tr> 						<tr><td>contigSizes</td><td>file</td><td>absolute</td><td></td><td>*</td><td>file containing the sizes of the contigs used in the BAM file if ranges should be extended (format: &lt;chrName&gt;&lt;TAB&gt;&lt;SIZE&gt;)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable60" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>wiggleFile</td><td>string</td><td>absolute path to the converted output file</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Bedtools (%SOFTWARE_VERSION%) was used to convert BAM files to WIG files.</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/20110278" data-uk-tooltip="title: PMID" target="_blank">20110278</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://bedtools.readthedocs.io/en/latest" data-uk-tooltip="title: Website" target="_blank">https://bedtools.readthedocs.io/en/latest</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-08-08" meta-search="true" meta-id="13">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SEQUENCING</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-08-08
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full13" onclick="initFullModule(13, 1)" uk-toggle>bamContigtDistribution</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">creates plots based on statistics of BAM files</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full13" class="uk-modal-full" meta-id="13" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>bamContigtDistribution</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch13">creates plots based on statistics of BAM files</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU R</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable13" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>bamMergedStats</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to the merged bam stats file</td><td>0</td><td>0</td></tr> 						<tr><td>outputFile</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to a output pdf file</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable13" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§SEQUENCING§" meta-update="2019-02-11" meta-search="true" meta-id="51">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full51" onclick="initFullModule(51, 1)" uk-toggle>bamToBed</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">converts bam format into bed format</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
					<a href="https://bedtools.readthedocs.io/en/latest/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/20110278" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full51" class="uk-modal-full" meta-id="51" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>bamToBed</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch51">converts bam format into bed format</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">bedtools (version 2.x)</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable51" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>inBam</td><td>string</td><td>valid file path, bam format</td><td></td><td>-1-</td><td>Path to the bam file that will be converted into bed format. An index of the bam file is not required.</td><td>0</td><td>0</td></tr> 						<tr><td>outBed</td><td>string</td><td></td><td></td><td>-1-</td><td>Path for saving the resulting bed file.</td><td>0</td><td>0</td></tr> 						<tr><td>bedtoolsPath</td><td>string</td><td>valid file path to executable</td><td>bedtools</td><td>-1-</td><td>Path to the bedtools executable. Per default, it is assumed that bedtools is in the PATH variable.</td><td>0</td><td>0</td></tr> 						<tr><td>split</td><td>boolean</td><td></td><td>true</td><td>-1-</td><td>Defines how split alignments (cigar string that contains N) are handled. If true, the skipped region is not included in the bed regions. If false, the skipped region is included in the bed region, i.e. there is only one interval from alignment start to alignment end.</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable51" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>bedFile</td><td>string</td><td>path to the bed file that is created (same value as outBed parameter)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Bed files were created from the bam files using bedtools bamtobed.</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/20110278" data-uk-tooltip="title: PMID" target="_blank">20110278</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://bedtools.readthedocs.io/en/latest/" data-uk-tooltip="title: Website" target="_blank">https://bedtools.readthedocs.io/en/latest/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§SEQUENCING§" meta-update="2019-02-11" meta-search="true" meta-id="61">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full61" onclick="initFullModule(61, 1)" uk-toggle>bamToBigWig</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">converts bam format into bigwig format</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
					<a href="https://deeptools.readthedocs.io/en/develop/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/27079975" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full61" class="uk-modal-full" meta-id="61" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>bamToBigWig</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch61">converts bam format into bigwig format</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">deeptools &gt;=2.0</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable61" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>inBam</td><td>string</td><td>valid file path, file ending .bam, indexed</td><td></td><td>-1-</td><td>Path to the bam file that will be converted into bigWig format. The bam file has to be indexed.</td><td>0</td><td>0</td></tr> 						<tr><td>outBw</td><td>string</td><td></td><td></td><td>-1-</td><td>Path for saving the resulting bigWig file.</td><td>0</td><td>0</td></tr> 						<tr><td>bamCoveragePath</td><td>string</td><td>valid file path</td><td>bamCoverage</td><td>-1-</td><td>Path to the executable bamCoverage which is part of deepTools. Per default, it is assumed that bamCoverage is in the PATH variable.</td><td>0</td><td>0</td></tr> 						<tr><td>binSize</td><td>integer</td><td>positive, not zero</td><td>1</td><td>-1-</td><td>Resolution of the bigWig file. Increasing the binSize causes loss of information but decreases the size of the bigWig file. Highest resolution (at single basepair level) is achieved for binSize=1 (default).</td><td>0</td><td>0</td></tr> 						<tr><td>numberOfProcessors</td><td>integer</td><td>positive, not zero</td><td>1</td><td>-1-</td><td>Number of processors to use (parallelization)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable61" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>bigWigFile</td><td>string</td><td>Path to the output bigWig file</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>BigWig files were created from the bam files using the tool bamCoverage from the deepTools tool suite.</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/27079975" data-uk-tooltip="title: PMID" target="_blank">27079975</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://deeptools.readthedocs.io/en/develop/" data-uk-tooltip="title: Website" target="_blank">https://deeptools.readthedocs.io/en/develop/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§;§QUALITY CONTROL§" meta-update="2019-03-15" meta-search="true" meta-id="17">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-03-15
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full17" onclick="initFullModule(17, 1)" uk-toggle>bamstats</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">creates various statistics on BAM files using RSeQC and samtools, which can be used for quality assessment</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="http://rseqc.sourceforge.net" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/22743226" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full17" class="uk-modal-full" meta-id="17" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>bamstats</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch17">creates various statistics on BAM files using RSeQC and samtools, which can be used for quality assessment</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">rseqc</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">samtools</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable17" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>bam</td><td>file path</td><td>absolute</td><td></td><td>1-</td><td>path to one or more BAM file(s)</td><td>0</td><td>0</td></tr> 						<tr><td>outdir</td><td>folder path</td><td>absolute</td><td></td><td>1</td><td>path to the output folder; individual files will be stored in a sub-folder (using the basename of the BAM file as folder name)</td><td>0</td><td>0</td></tr> 						<tr><td>readLength</td><td>integer</td><td></td><td></td><td>1</td><td>maximal length of the reads</td><td>0</td><td>0</td></tr> 						<tr><td>sampleDepth</td><td>integer</td><td></td><td>100000</td><td>*</td><td>number of reads which are used for sampling</td><td>0</td><td>0</td></tr> 						<tr><td>annotation</td><td>file path</td><td>absolute</td><td></td><td>*</td><td>gene annotation in BED format</td><td>0</td><td>0</td></tr> 						<tr><td>geneBodyAnnotation</td><td>file path</td><td>absolute</td><td></td><td>*</td><td>genes that are used to calculate the gene body coverage; should contain house keeping genes</td><td>0</td><td>0</td></tr> 						<tr><td>idxstats</td><td>boolean</td><td></td><td>true</td><td>*</td><td>enables calculation of number of reads mapped on each chromosome</td><td>0</td><td>0</td></tr> 						<tr><td>flagstat</td><td>boolean</td><td></td><td>true</td><td>*</td><td>enables calculation of flags of mapped reads</td><td>0</td><td>0</td></tr> 						<tr><td>count</td><td>boolean</td><td></td><td>false</td><td>*</td><td>enables calculation of raw and rpkm count table for exons, introns and mRNAs</td><td>0</td><td>0</td></tr> 						<tr><td>saturation</td><td>boolean</td><td></td><td>true</td><td>*</td><td>enables down-sampling of the mapped reads to infer the sequencing depth</td><td>0</td><td>0</td></tr> 						<tr><td>statistics</td><td>boolean</td><td></td><td>true</td><td>*</td><td>calculates reads mapping statistics</td><td>0</td><td>0</td></tr> 						<tr><td>clipping</td><td>boolean</td><td></td><td>true</td><td>*</td><td>enables clipping statistic of the mapped reads</td><td>0</td><td>0</td></tr> 						<tr><td>insertion</td><td>boolean</td><td></td><td>true</td><td>*</td><td>enables insertion statistic of the mapped reads</td><td>0</td><td>0</td></tr> 						<tr><td>deletion</td><td>boolean</td><td></td><td>true</td><td>*</td><td>enables deletion statistic of the mapped reads</td><td>0</td><td>0</td></tr> 						<tr><td>inferExperiment</td><td>boolean</td><td></td><td>true</td><td>*</td><td>tries to infer if the sequencing was strand specific or not</td><td>0</td><td>0</td></tr> 						<tr><td>junctionAnnotation</td><td>boolean</td><td></td><td>true</td><td>*</td><td>enables checking of how many of the splice junctions are novel or annotated</td><td>0</td><td>0</td></tr> 						<tr><td>junctionSaturation</td><td>boolean</td><td></td><td>true</td><td>*</td><td>enables down-sampling of the spliced reads to infer if sequencing depth is enough for splicing analyses</td><td>0</td><td>0</td></tr> 						<tr><td>distribution</td><td>boolean</td><td></td><td>true</td><td>*</td><td>calculates how mapped reads are distributed among different genomic features</td><td>0</td><td>0</td></tr> 						<tr><td>duplication</td><td>boolean</td><td></td><td>true</td><td>*</td><td>calculates sequence duplication levels</td><td>0</td><td>0</td></tr> 						<tr><td>gc</td><td>boolean</td><td></td><td>true</td><td>*</td><td>calculates GC-content of the mapped reads</td><td>0</td><td>0</td></tr> 						<tr><td>nvc</td><td>boolean</td><td></td><td>true</td><td>*</td><td>checks if a nucleotide composition bias exist</td><td>0</td><td>0</td></tr> 						<tr><td>insertSize</td><td>boolean</td><td></td><td>true</td><td>*</td><td>calculates the insert size between two paired RNA reads</td><td>0</td><td>0</td></tr> 						<tr><td>fragmentSize</td><td>boolean</td><td></td><td>true</td><td>*</td><td>calculates the fragment size for each transcript</td><td>0</td><td>0</td></tr> 						<tr><td>tin</td><td>boolean</td><td></td><td>true</td><td>*</td><td>calculates the transcript integrity number which is similar to the RNA integrity number</td><td>0</td><td>0</td></tr> 						<tr><td>paired</td><td>boolean</td><td></td><td>false</td><td>*</td><td>must be set if paired-end data is analyzed</td><td>0</td><td>0</td></tr> 						<tr><td>stranded</td><td>boolean</td><td></td><td>false</td><td>*</td><td>must be set if strand-specific data is analyzed</td><td>0</td><td>0</td></tr> 						<tr><td>disableAllDefault</td><td>boolean</td><td></td><td>false</td><td>*</td><td>disables all options which are not explicitly activated</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable17" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Quality of the resulting mappings was assessed using RSeQC [Liguo Wang, Shengqin Wang, Wei Li; RSeQC: quality control of RNA-seq experiments, Bioinformatics, Volume 28, Issue 16, 15 August 2012, Pages 2184–2185].</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/22743226" data-uk-tooltip="title: PMID" target="_blank">22743226</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="http://rseqc.sourceforge.net" data-uk-tooltip="title: Website" target="_blank">http://rseqc.sourceforge.net</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§;§GENE EXPRESSION§" meta-update="2019-03-15" meta-search="true" meta-id="20">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-03-15
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full20" onclick="initFullModule(20, 1)" uk-toggle>bedgraphReplicateMerger</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">combines expression of biological or technical replicates; all replicates are scaled to the same number of reads and averaged afterwards</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full20" class="uk-modal-full" meta-id="20" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>bedgraphReplicateMerger</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch20">combines expression of biological or technical replicates; all replicates are scaled to the same number of reads and averaged afterwards</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable20" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>bedgraphFiles</td><td>file path</td><td>absolute</td><td></td><td>2-</td><td>path to sorted BEDGRAPH files; at least two files must be given; all files must contain the same chromosomes in the same order</td><td>0</td><td>0</td></tr> 						<tr><td>outputFile</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to the output file</td><td>0</td><td>0</td></tr> 						<tr><td>mergedIdxstatsFile</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to a tab-separated file that contains the output generated by samtools idxstats for all samples (columns 1-4) and in the 5th column the sample name; used columns: 1 -&gt; chr name; 3 -&gt; number of mapped reads; 5 -&gt; name of the sample</td><td>0</td><td>0</td></tr> 						<tr><td>notSkipHead</td><td>boolean</td><td></td><td>false</td><td>*</td><td>disables the skipping of the first line of the idxstats file (--mergedIdxstatsFile); default: first line is skipped</td><td>0</td><td>0</td></tr> 						<tr><td>numberOfDigits</td><td>integer</td><td></td><td>5</td><td>*</td><td>number of decimal places to round the calculated values</td><td>0</td><td>0</td></tr> 						<tr><td>normByReadCount</td><td>integer</td><td></td><td>1000000</td><td>*</td><td>number of reads to which each replicate is normed (based on the idxstats output) before values are averaged</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable20" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-08-13" meta-search="true" meta-id="36">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SEQUENCING</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-08-13
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full36" onclick="initFullModule(36, 1)" uk-toggle>bedgraphShrinker</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">shrinks regions with the same score to one region in a bedgraph file or expands the file to a region size of one basepair</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full36" class="uk-modal-full" meta-id="36" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>bedgraphShrinker</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch36">shrinks regions with the same score to one region in a bedgraph file or expands the file to a region size of one basepair</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable36" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>bedgraphFile</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to a sorted, not overlapping BEDGRAPH file</td><td>0</td><td>0</td></tr> 						<tr><td>outputFile</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to the output file</td><td>0</td><td>0</td></tr> 						<tr><td>genomeSize</td><td>file</td><td>absolute</td><td></td><td>*</td><td>path to file containing the size of the contigs</td><td>0</td><td>0</td></tr> 						<tr><td>expand</td><td>boolean</td><td></td><td>false</td><td>*</td><td>expand the ranges instead of shrinking them</td><td>0</td><td>0</td></tr> 						<tr><td>addZeroRanges</td><td>boolean</td><td></td><td>false</td><td>*</td><td>adds ranges that are missing with a zero value</td><td>0</td><td>0</td></tr> 						<tr><td>omitZeroRanges</td><td>boolean</td><td></td><td>false</td><td>*</td><td>suppress the output of ranges with a zero value</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable36" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§;§TECHNICAL DEMO§" meta-update="2019-03-13" meta-search="true" meta-id="40">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-03-13
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full40" onclick="initFullModule(40, 1)" uk-toggle>bowtie2Docker</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">technical demo which shows how docker containers can be used in combination with Watchdog; basic bowtie mapper</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://quay.io/repository/biocontainers/bowtie2" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full40" class="uk-modal-full" meta-id="40" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>bowtie2Docker</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch40">technical demo which shows how docker containers can be used in combination with Watchdog; basic bowtie mapper</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">docker</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable40" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>genome</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to indexed reference genome (withouth trailing .X.bt2 ending)</td><td>0</td><td>0</td></tr> 						<tr><td>reads</td><td>file path</td><td>absolute</td><td></td><td>*</td><td>path to reads in FASTQ format; for mapping of paired-end data, two files are required</td><td>0</td><td>0</td></tr> 						<tr><td>outfile</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to output file, which is written in SAM format; a log file with .log suffix will also be written</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable40" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://quay.io/repository/biocontainers/bowtie2" data-uk-tooltip="title: Website" target="_blank">https://quay.io/repository/biocontainers/bowtie2</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat3-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§CHIP-SEQ§" meta-update="2019-02-11" meta-search="true" meta-id="34">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">ChIP-seq</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full34" onclick="initFullModule(34, 1)" uk-toggle>bwaAln</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">maps reads with bwa aln</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
					<a href="http://bio-bwa.sourceforge.net" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/19451168" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full34" class="uk-modal-full" meta-id="34" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>bwaAln</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch34">maps reads with bwa aln</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">bwa</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable34" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>inReads</td><td>file</td><td>file exists, fastq format, ending .fq or .fastq</td><td></td><td>1</td><td>fastq file with the sequenced reads</td><td>0</td><td>0</td></tr> 						<tr><td>bwaIndex</td><td>string</td><td></td><td></td><td>1</td><td>Common prefix of bwa index files for the reference genome</td><td>0</td><td>0</td></tr> 						<tr><td>outSai</td><td>string</td><td></td><td></td><td>1</td><td>file for writing mapped reads in bwa format</td><td>0</td><td>0</td></tr> 						<tr><td>bwaPath</td><td>file</td><td>file exists, bwa executable</td><td>bwa</td><td>*</td><td>path to BWA executable (default: use executable from PATH)</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td>&gt;0</td><td>1</td><td>*</td><td>number of threads to use for bwa aln (-t option of bwa)</td><td>0</td><td>0</td></tr> 						<tr><td>stopIfMoreThanBestHits</td><td>integer</td><td>&gt;0</td><td></td><td>*</td><td>stop searching when there are more than that many best hits (default: use bwa default)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable34" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>bwaSaiFile</td><td>string</td><td>*.sai file created by the module (same value as given by the parameter outSai)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>We mapped the reads with bwa aln.</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/19451168" data-uk-tooltip="title: PMID" target="_blank">19451168</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="http://bio-bwa.sourceforge.net" data-uk-tooltip="title: Website" target="_blank">http://bio-bwa.sourceforge.net</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat3-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§CHIP-SEQ§" meta-update="2019-02-11" meta-search="true" meta-id="57">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">ChIP-seq</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full57" onclick="initFullModule(57, 1)" uk-toggle>bwaSampe</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">creates a sam file with bwa sampe from mappings of bwa aln for paired reads</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
					<a href="http://bio-bwa.sourceforge.net" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/19451168" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full57" class="uk-modal-full" meta-id="57" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>bwaSampe</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch57">creates a sam file with bwa sampe from mappings of bwa aln for paired reads</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">bwa</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable57" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>inReads1</td><td>file</td><td>file exists, fastq format, ending .fq or .fastq</td><td></td><td>1</td><td>uncompressed fastq (.fq, .fastq) file with the sequenced reads</td><td>0</td><td>0</td></tr> 						<tr><td>inReads2</td><td>file</td><td>file exists, fastq format, ending .fq or .fastq</td><td></td><td>1</td><td>uncompressed fastq (.fq, .fastq) file with the sequenced reads (mates)</td><td>0</td><td>0</td></tr> 						<tr><td>inSai1</td><td>file</td><td>file exists</td><td></td><td>1</td><td>output of bwa aln for the file given by inReads1</td><td>0</td><td>0</td></tr> 						<tr><td>inSai2</td><td>file</td><td>file exists</td><td></td><td>1</td><td>output of bwa aln for the file given by inReads2</td><td>0</td><td>0</td></tr> 						<tr><td>bwaIndex</td><td>string</td><td></td><td></td><td>1</td><td>Common prefix of bwa index files for the reference genome</td><td>0</td><td>0</td></tr> 						<tr><td>outSam</td><td>string</td><td></td><td></td><td>1</td><td>file for writing mapped reads in sam format</td><td>0</td><td>0</td></tr> 						<tr><td>bwaPath</td><td>file</td><td>file exists, bwa executable</td><td>bwa</td><td>*</td><td>path to BWA executable (default: use executable from PATH)</td><td>0</td><td>0</td></tr> 						<tr><td>indexInRam</td><td>boolean</td><td></td><td>False</td><td>*</td><td>option to load complete index into main memory (default: false)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable57" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>bwaPairedSamFile</td><td>string</td><td>sam file created by the module (same value as given by the parameter outSam)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>We created a sam file with bwa sampe.</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/19451168" data-uk-tooltip="title: PMID" target="_blank">19451168</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="http://bio-bwa.sourceforge.net" data-uk-tooltip="title: Website" target="_blank">http://bio-bwa.sourceforge.net</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat2-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§GENERAL§" meta-update="2018-10-31" meta-search="true" meta-id="4">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">General</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2018-10-31
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full4" onclick="initFullModule(4, 1)" uk-toggle>checksum</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">creates a md5 checksum of a file or verifies file integrity based on a md5 checksum using md5sum</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://www.gnu.org/software/coreutils/coreutils.html" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full4" class="uk-modal-full" meta-id="4" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>checksum</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch4">creates a md5 checksum of a file or verifies file integrity based on a md5 checksum using md5sum</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU md5sum</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable4" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>input</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>absolute path to file for which a checksum should be calculated or which should be verified</td><td>0</td><td>0</td></tr> 						<tr><td>oldChecksumName</td><td>file path</td><td>absolute</td><td></td><td>*</td><td>absolute path to a (non-existent) file used to identify the correct checksum line for cases in which the file was renamed or moved after checksum creation; can only be used in verify mode</td><td>0</td><td>0</td></tr> 						<tr><td>checksum</td><td>file path</td><td>absolute</td><td>.checksum.md5</td><td>*</td><td>absolute path to the checksum file; by default '.checksum.md5' located in the same directory as the input file</td><td>0</td><td>0</td></tr> 						<tr><td>verify</td><td>boolean</td><td></td><td>false</td><td>*</td><td>flag to verify integrity of a file based on the checksum file</td><td>0</td><td>0</td></tr> 						<tr><td>update</td><td>boolean</td><td></td><td>false</td><td>*</td><td>flag to update an already existing checksum in the checksum file</td><td>0</td><td>0</td></tr> 						<tr><td>absolutePath</td><td>boolean</td><td></td><td>false</td><td>*</td><td>flag to store an absolute path in the checksum file instead of a relative one</td><td>0</td><td>0</td></tr> 						<tr><td>ignorePath</td><td>boolean</td><td></td><td>false</td><td>*</td><td>flag to use only the name of the file for identification of the corresponding checksum line (ignores the location of the file); can only be used in verify mode</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable4" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.gnu.org/software/coreutils/coreutils.html" data-uk-tooltip="title: Website" target="_blank">https://www.gnu.org/software/coreutils/coreutils.html</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat4-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§CIRCRNA§" meta-update="2019-02-11" meta-search="true" meta-id="39">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">circRNA</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full39" onclick="initFullModule(39, 1)" uk-toggle>circCombination</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">combines the predictions of circularRNAs made with the modules for CIRI2 and circRNA_finder.</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full39" class="uk-modal-full" meta-id="39" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>circCombination</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch39">combines the predictions of circularRNAs made with the modules for CIRI2 and circRNA_finder.</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">Python3</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable39" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>inCircs1</td><td>file</td><td>file exists</td><td></td><td>1</td><td>First prediction file with circRNAs and junction reads (tab-separated, 5 columns: chromosome, start, end, strand, list of reads)</td><td>0</td><td>0</td></tr> 						<tr><td>inCircs2</td><td>file</td><td>file exists</td><td></td><td>1</td><td>Second prediction file with circRNAs and junction reads (tab-separated, 5 columns: chromosome, start, end, strand, list of reads)</td><td>0</td><td>0</td></tr> 						<tr><td>outUnion</td><td>file</td><td></td><td></td><td>1</td><td>Output path for the union of the predictions (coordinates and reads)</td><td>0</td><td>0</td></tr> 						<tr><td>outIntersection</td><td>file</td><td></td><td></td><td>1</td><td>Output path for the intersection of the predictions (coordinates and reads</td><td>0</td><td>0</td></tr> 						<tr><td>outIntersectedUnion</td><td>file</td><td></td><td></td><td>1</td><td>Output path for the intersected union of the predictions (intersection of coordinates, union of reads)</td><td>0</td><td>0</td></tr> 						<tr><td>minReads</td><td>integer</td><td>&gt;=1</td><td>2</td><td>*</td><td>
        Minimum number of predicted junction reads required for writing a circRNA into the output files. The cutoff is applied independently to the intersection, union and intersected union of the predictions.
      </td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable39" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>circUnion</td><td>file</td><td>Output path for the union of the predictions (same as input parameter outUnion)</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>circIntersection</td><td>file</td><td>Output path for the intersection of the predictions (same as input parameter outIntersection)</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>circIntersectedUnion</td><td>file</td><td>Output path for the intersected union of the predictions (same as input parameter outIntersectedUnion)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>
        Predictions of circular RNAs were combined by forming the union/intersection of the individual predictions. The circular reads were combined by forming the union/intersection of the predictions.
    </p>
					<p>Pubmed references:

						
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat4-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§CIRCRNA§" meta-update="2019-02-11" meta-search="true" meta-id="19">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">circRNA</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full19" onclick="initFullModule(19, 1)" uk-toggle>circRNAfinder</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">runs circRNA_finder to detect circular RNAs in single-end or paired-end sequencing data.</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
					<a href="https://github.com/orzechoj/circRNA_finder" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/25544350" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full19" class="uk-modal-full" meta-id="19" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>circRNAfinder</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch19">runs circRNA_finder to detect circular RNAs in single-end or paired-end sequencing data.</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">perl</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">awk</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">samtools</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable19" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>inReads1</td><td>file</td><td>file exists, fastq format</td><td></td><td>*</td><td>path to single-end fastq file or path to first fastq file with paired reads</td><td>0</td><td>0</td></tr> 						<tr><td>inReads2</td><td>file</td><td>file exists, fastq format</td><td></td><td>*</td><td>path to second fastq file with paired reads (paired-end data only)</td><td>0</td><td>0</td></tr> 						<tr><td>strandedLibrary</td><td>integer</td><td>allowed values: 0,1,2</td><td>0</td><td>*</td><td>indicates if the library is strand specific, 0 = unstranded/unknown, 1 = stranded (first read), 2 = stranded (second read), (default: 0),if the library type is unstranded/unknown the strand is guessed from the strand of the AG-GT splice site</td><td>0</td><td>0</td></tr> 						<tr><td>reference</td><td>file</td><td>file exists, fasta format</td><td></td><td>*</td><td>path to (multi-)fasta file with the reference genome (not required if STAR index or a STAR results is provided)</td><td>0</td><td>0</td></tr> 						<tr><td>inSTAR</td><td>string</td><td></td><td></td><td>*</td><td>output prefix of a STAR mapping that was created with STAR run with chimeric segment detection</td><td>0</td><td>0</td></tr> 						<tr><td>outPrefix</td><td>string</td><td></td><td></td><td>1</td><td>path and file name prefix for all files produced by this module; the final file is named out/prefixcfCirc.txt</td><td>0</td><td>0</td></tr> 						<tr><td>outCirc</td><td>string</td><td></td><td></td><td>*</td><td>final output of predicted CircRNAs (can be used to save the final prediction in a different place than given in outPrefix)</td><td>0</td><td>0</td></tr> 						<tr><td>starPath</td><td>file</td><td></td><td>STAR</td><td>*</td><td>specify a path to the STAR executable if STAR is not part of your PATH variable</td><td>0</td><td>0</td></tr> 						<tr><td>starIndex</td><td>file</td><td></td><td></td><td>*</td><td>STAR index for the reference genome, if no index is provided it is automatically created by the module using the file given by --reference</td><td>0</td><td>0</td></tr> 						<tr><td>starThreads</td><td>integer</td><td>&gt;=1</td><td>1</td><td>*</td><td>number of threads to use with STAR</td><td>0</td><td>0</td></tr> 						<tr><td>cfPath</td><td>file</td><td></td><td>postProcessStarAlignment.pl</td><td>*</td><td>path to circRNA_finder perl script postProcessStarAlignment.pl</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable19" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>cfCircs</td><td>string</td><td>path to file with predicted circRNAs, it corresponds to the value of the parameter outCirc if it is set, otherwise the file path is derived from outPrefix</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>We predicted circular RNAs using circRNA_finder.</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/25544350" data-uk-tooltip="title: PMID" target="_blank">25544350</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://github.com/orzechoj/circRNA_finder" data-uk-tooltip="title: Website" target="_blank">https://github.com/orzechoj/circRNA_finder</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat4-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§CIRCRNA§" meta-update="2019-02-11" meta-search="true" meta-id="26">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">circRNA</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full26" onclick="initFullModule(26, 1)" uk-toggle>ciri2</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">runs CIRI2 to detect circular RNAs in single-end or paired-end sequencing data.</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
					<a href="https://sourceforge.net/projects/ciri/files/CIRI2/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/28334140" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full26" class="uk-modal-full" meta-id="26" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>ciri2</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch26">runs CIRI2 to detect circular RNAs in single-end or paired-end sequencing data.</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">perl</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable26" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>inReads1</td><td>file</td><td>file exists, fastq format</td><td></td><td>*</td><td>path to first fastq file with reads (for single-end or paired-end data first reads)</td><td>0</td><td>0</td></tr> 						<tr><td>inReads2</td><td>file</td><td>file exists, fastq format</td><td></td><td>*</td><td>path to second fastq file with reads (for paired-end data second reads only)</td><td>0</td><td>0</td></tr> 						<tr><td>inSAM</td><td>file</td><td>file exists, SAM format</td><td></td><td>*</td><td>path to SAM file that was created with BWA Mem (can be used as input instead of fastq files)</td><td>0</td><td>0</td></tr> 						<tr><td>reference</td><td>file</td><td>file exists, fasta format</td><td></td><td>1</td><td>path to (multi-)fasta file with the reference genome</td><td>0</td><td>0</td></tr> 						<tr><td>outPrefix</td><td>string</td><td></td><td></td><td>1</td><td>path and file name prefix for all files produced by this module, the final file is named out/prefixciriCirc.txt</td><td>0</td><td>0</td></tr> 						<tr><td>outCirc</td><td>string</td><td></td><td></td><td>*</td><td>final output of predicted CircRNAs (can be used to save the final prediction in a different place than given in outPrefix)</td><td>0</td><td>0</td></tr> 						<tr><td>bwaPath</td><td>file</td><td></td><td>bwa</td><td>*</td><td>specify a path to the BWA executable if bwa is not part of your PATH variable</td><td>0</td><td>0</td></tr> 						<tr><td>bwaThreads</td><td>integer</td><td>&gt;=1</td><td>1</td><td>*</td><td>number of threads to use with BWA, default:1</td><td>0</td><td>0</td></tr> 						<tr><td>bwaIndex</td><td>string</td><td>valid bwa index</td><td></td><td>*</td><td>BWA index for the reference genome provided by the --reference option, if no index is provided it is automatically created by the module</td><td>0</td><td>0</td></tr> 						<tr><td>bwaSeedSize</td><td>integer</td><td>&gt;=1</td><td>19</td><td>*</td><td>BWA -k parameter for the minimum seed length</td><td>0</td><td>0</td></tr> 						<tr><td>bwaScoreThreshold</td><td>integer</td><td>&gt;=1</td><td>30</td><td>*</td><td>BWA -T parameter for the minimum alignment score; default is 30, but 19 recommended for CIRI2</td><td>0</td><td>0</td></tr> 						<tr><td>ciriPath</td><td>file</td><td></td><td>CIRI2.pl</td><td>*</td><td>path to CIRI2 perl script</td><td>0</td><td>0</td></tr> 						<tr><td>ciriThreads</td><td>integer</td><td>&gt;=1</td><td>1</td><td>*</td><td>number of threads to use for CIRI2</td><td>0</td><td>0</td></tr> 						<tr><td>ciriAnnotation</td><td>file</td><td>file exists, GTF format</td><td></td><td>*</td><td>GTF file with gene annotations for the genome given in the --reference option, if a GTF file is passed to this module, CIRI annotates all circRNAs with the corresponding gene</td><td>0</td><td>0</td></tr> 						<tr><td>ciriStringency</td><td>string</td><td>3 allowed values: high, medium or low</td><td>high</td><td>*</td><td>Controls how stringent CIRI2 filters the circRNAs based on circular reads, cigar strings and false positive reads</td><td>0</td><td>0</td></tr> 						<tr><td>ciriKeepTmpFiles</td><td>boolean</td><td></td><td>False</td><td>*</td><td>if this flag is set, CIRI2 does not delete the temporary files at the end</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable26" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>ciriCircs</td><td>string</td><td>path to file with predicted circRNAs, it corresponds to the value of the parameter outCirc if it is set, otherwise the file path is derived from outPrefix</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>We predicted circular RNAs using CIRI2.</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/28334140" data-uk-tooltip="title: PMID" target="_blank">28334140</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://sourceforge.net/projects/ciri/files/CIRI2/" data-uk-tooltip="title: Website" target="_blank">https://sourceforge.net/projects/ciri/files/CIRI2/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat5-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§FILE UTILS§" meta-update="2019-02-11" meta-search="true" meta-id="9">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">File Utils</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full9" onclick="initFullModule(9, 1)" uk-toggle>concatenateFiles</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">concatenates 2 or more files.</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full9" class="uk-modal-full" meta-id="9" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>concatenateFiles</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch9">concatenates 2 or more files.</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">Python3</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable9" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>inFile</td><td>file</td><td>file exists</td><td></td><td>*</td><td>input files given in the order of concatenation, files with ending .gz are interpreted as compressed files and are extracted</td><td>0</td><td>0</td></tr> 						<tr><td>outFile</td><td>file</td><td></td><td></td><td>1</td><td>path to save the concatenated files</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable9" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>concatenatedFile</td><td>string</td><td>path of the concatenated file, this is the same value as given by the parameter outFile</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-08-08" meta-search="true" meta-id="54">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SEQUENCING</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-08-08
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full54" onclick="initFullModule(54, 1)" uk-toggle>contextMap</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">context-based RNA-seq read mapping</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://www.bio.ifi.lmu.de/software/contextmap/index.html" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/25928589" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full54" class="uk-modal-full" meta-id="54" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>contextMap</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch54">context-based RNA-seq read mapping</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">java</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">bwa / bowtie / bowtie2</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU core utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable54" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>jarPath</td><td>file</td><td>absolute</td><td></td><td>*</td><td>path to ContextMap jar file; if not given internal version will be used</td><td>0</td><td>0</td></tr> 						<tr><td>reads</td><td>file</td><td>absolute</td><td></td><td>*</td><td>path to reads in fasta or fastq format</td><td>0</td><td>0</td></tr> 						<tr><td>alignerName</td><td>enum</td><td></td><td></td><td>1</td><td>name of short-read alignment tool; supported values: 'bwa', 'bowtie1' or 'bowtie2'</td><td>0</td><td>0</td></tr> 						<tr><td>alignerBin</td><td>file</td><td>absolute</td><td></td><td>*</td><td>path to the executable of the chosen aligner tool</td><td>0</td><td>0</td></tr> 						<tr><td>indexerBin</td><td>file</td><td>absolute</td><td></td><td>*</td><td>path to the executable of the aligner's indexing tool (not needed for BWA)</td><td>0</td><td>0</td></tr> 						<tr><td>indices</td><td>file</td><td>absolute</td><td></td><td>1-</td><td>comma separated list of paths to basenames of indices, which can be used by the chosen aligner</td><td>0</td><td>0</td></tr> 						<tr><td>genome</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to a directory with genome sequences in fasta format (each chromosome in a separate file)</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to the output directory</td><td>0</td><td>0</td></tr> 						<tr><td>skipsplit</td><td>string</td><td></td><td></td><td>*</td><td>comma separated list of booleans, each element refers to a given aligner index (same ordering); 'true' for no split detection, 'false' otherwise (req. in mining mode).</td><td>0</td><td>0</td></tr> 						<tr><td>skipmultisplit</td><td>string</td><td></td><td></td><td>*</td><td>comma separated list of booleans, each element refers to a given aligner index (same ordering); 'true' for no multisplit detection, 'false' otherwise (req. in mining mode).</td><td>0</td><td>0</td></tr> 						<tr><td>speciesindex</td><td>string</td><td></td><td></td><td>*</td><td>path to a directory containing index files created with the 'indexer' tool (req. in mining mode)</td><td>0</td><td>0</td></tr> 						<tr><td>alignerTmp</td><td>string</td><td></td><td></td><td>*</td><td>path to a directory for temporary alignment files</td><td>0</td><td>0</td></tr> 						<tr><td>seed</td><td>integer</td><td>&gt;0</td><td></td><td>*</td><td>seed length for the alignment (default: Bwt1: 30, BWA/Bwt2: 20)</td><td>0</td><td>0</td></tr> 						<tr><td>splitseedsizes</td><td>integer</td><td>&gt;0</td><td>15</td><td>*</td><td>seed size for the split search seed (default: 15)</td><td>0</td><td>0</td></tr> 						<tr><td>mismatches</td><td>integer</td><td>&gt;=0</td><td>4</td><td>*</td><td>allowed mismatches in the whole read</td><td>0</td><td>0</td></tr> 						<tr><td>seedmismatches</td><td>integer</td><td>&gt;=0</td><td></td><td>*</td><td>allowed mismatches in the seed region (default: Bwt1: 1, BWA/Bwt2: 0)</td><td>0</td><td>0</td></tr> 						<tr><td>splitseedmismatches</td><td>integer</td><td>&gt;=0</td><td>0</td><td>*</td><td>allowed mismatches for the split seed search (default: 0)</td><td>0</td><td>0</td></tr> 						<tr><td>mmdiff</td><td>integer</td><td>&gt;=1</td><td>0</td><td>*</td><td>maximum allowed mismatch difference between the best and second best alignment of the same read</td><td>0</td><td>0</td></tr> 						<tr><td>maxhits</td><td>integer</td><td>&gt;=1</td><td></td><td>*</td><td>maximum number of candidate alignments per read; reads with more hits are skipped (bwa/bwt1) or the already found hits are reported (bwt2) (default for bwa/bwt1:10, bwt2: 3)</td><td>0</td><td>0</td></tr> 						<tr><td>minsize</td><td>integer</td><td>&gt;=1</td><td>10</td><td>*</td><td>minimum number of reads a genomic region has to contain for being regarded as a local context</td><td>0</td><td>0</td></tr> 						<tr><td>maxindelsize</td><td>integer</td><td>&gt;=0</td><td>10</td><td>*</td><td>maximum allowed size of insertions or deletions (default: 10)</td><td>0</td><td>0</td></tr> 						<tr><td>gtf</td><td>file</td><td>absolute</td><td></td><td>*</td><td>path to an annotation file in gtf format</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>string</td><td></td><td></td><td>*</td><td>number of threads used for mapping</td><td>0</td><td>0</td></tr> 						<tr><td>localTmpFolder</td><td>folder</td><td>absolute</td><td>/usr/local/storage/</td><td>*</td><td>path to a local storage that is used for temporary data</td><td>0</td><td>0</td></tr> 						<tr><td>mining</td><td>boolean</td><td></td><td>false</td><td>*</td><td>enables the mining for infections or contaminations</td><td>0</td><td>0</td></tr> 						<tr><td>noclipping</td><td>boolean</td><td></td><td>false</td><td>*</td><td>disables the calculation of clipped alignments</td><td>0</td><td>0</td></tr> 						<tr><td>noncanonicaljunctions</td><td>boolean</td><td></td><td>false</td><td>*</td><td>enables the prediction of non-canonical splice sites</td><td>0</td><td>0</td></tr> 						<tr><td>strandspecific</td><td>boolean</td><td></td><td>false</td><td>*</td><td>enables strand specific mapping</td><td>0</td><td>0</td></tr> 						<tr><td>pairedend</td><td>boolean</td><td></td><td>false</td><td>*</td><td>enables mapping of paired-end reads; nomenclature for mates from the same fragment: base_name/1 and base_name/2, respectively; only valid for versions smaller than 2.7.2</td><td>0</td><td>0</td></tr> 						<tr><td>polyA</td><td>boolean</td><td></td><td>false</td><td>*</td><td>enables the search for polyA-tails (mutually exclusive with --noclipping)</td><td>0</td><td>0</td></tr> 						<tr><td>verbose</td><td>boolean</td><td></td><td>false</td><td>*</td><td>verbose mode</td><td>0</td><td>0</td></tr> 						<tr><td>keeptmp</td><td>boolean</td><td></td><td>false</td><td>*</td><td>does not delete some temporary files</td><td>0</td><td>0</td></tr> 						<tr><td>sequenceDB</td><td>boolean</td><td></td><td>false</td><td>*</td><td>sequence mapping to disk; recommended for very large data sets.</td><td>0</td><td>0</td></tr> 						<tr><td>memoryScaleFactor</td><td>integer</td><td>[0,100]</td><td>75</td><td>*</td><td>scale factor in percent that defines the proportion of the memory that is used for java; default memory: 3GB*threads*(scaleFactor/100)</td><td>0</td><td>0</td></tr> 						<tr><td>memoryPerThread</td><td>integer</td><td></td><td>3072</td><td>*</td><td>total memory per thread in MB if running on local host; otherwise memory limit of Watchdog executor might be set; default: 3072</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable54" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>contextMapSAMFile</td><td>string</td><td>path to mapped SAM file</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>contextMapPolyAFile</td><td>string</td><td>path to detected polyA tails</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>RNA-seq reads were mapped against the XXX genome using ContextMap (%SOFTWARE_VERSION%) with BWA as short read aligner and default parameters.</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/25928589" data-uk-tooltip="title: PMID" target="_blank">25928589</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.bio.ifi.lmu.de/software/contextmap/index.html" data-uk-tooltip="title: Website" target="_blank">https://www.bio.ifi.lmu.de/software/contextmap/index.html</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat5-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§FILE UTILS§" meta-update="2019-02-11" meta-search="true" meta-id="58">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">File Utils</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full58" onclick="initFullModule(58, 1)" uk-toggle>copyFile</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">copies a given file to a new location.</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full58" class="uk-modal-full" meta-id="58" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>copyFile</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch58">copies a given file to a new location.</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">Python3</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable58" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>sourcePath</td><td>file</td><td>file exists</td><td></td><td>1</td><td>path of the file to copy</td><td>0</td><td>0</td></tr> 						<tr><td>targetPath</td><td>file</td><td></td><td></td><td>1</td><td>path of the new location of the file, all non-existing parent directories of the file are created</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable58" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat5-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§FILE UTILS§" meta-update="2019-02-11" meta-search="true" meta-id="45">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">File Utils</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full45" onclick="initFullModule(45, 1)" uk-toggle>createFolder</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">creates a folder and its parent directories.</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full45" class="uk-modal-full" meta-id="45" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>createFolder</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch45">creates a folder and its parent directories.</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">Python3</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable45" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>folderPath</td><td>folder</td><td></td><td></td><td>1</td><td>folder that will be created</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable45" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-03-13" meta-search="true" meta-id="3">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-03-13
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full3" onclick="initFullModule(3, 1)" uk-toggle>cutadapt</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">sequence adapters can be removed and sequences can be trimmed based on length or base-call quality scores</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://github.com/marcelm/cutadapt" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full3" class="uk-modal-full" meta-id="3" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>cutadapt</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch3">sequence adapters can be removed and sequences can be trimmed based on length or base-call quality scores</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">cutadapt</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable3" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>fastq</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to one FASTQ file</td><td>0</td><td>0</td></tr> 						<tr><td>prim3</td><td>string</td><td></td><td></td><td>*</td><td>adapter that was ligated at the 3' end; '$' at the end will cause that the adapter is anchored at the end of the read</td><td>0</td><td>0</td></tr> 						<tr><td>prim5</td><td>string</td><td></td><td></td><td>*</td><td>adapter that was ligated at the 5' end; '^' at the start will cause that the adapter is anchored at the beginning of the read</td><td>0</td><td>0</td></tr> 						<tr><td>adapter</td><td>string</td><td></td><td></td><td>*</td><td>adapter that can be located at the 3' and 5' end</td><td>0</td><td>0</td></tr> 						<tr><td>errorRate</td><td>double</td><td>[0, 1]</td><td>0.05</td><td>*</td><td>maximum allowed error rate</td><td>0</td><td>0</td></tr> 						<tr><td>repeat</td><td>integer</td><td>[1, 100]</td><td>1</td><td>*</td><td>try to remove adapters at most N times</td><td>0</td><td>0</td></tr> 						<tr><td>minOverlap</td><td>integer</td><td>&gt;0</td><td>6</td><td>*</td><td>minimum overlap length</td><td>0</td><td>0</td></tr> 						<tr><td>minLength</td><td>integer</td><td>[1, 100000]</td><td>40</td><td>*</td><td>minimum read length after trimming</td><td>0</td><td>0</td></tr> 						<tr><td>maxLength</td><td>integer</td><td>[1, 100000]</td><td>-1</td><td>*</td><td>maximum read length after trimming</td><td>0</td><td>0</td></tr> 						<tr><td>outfile</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to an output file</td><td>0</td><td>0</td></tr> 						<tr><td>infofile</td><td>file path</td><td>absolute</td><td></td><td>*</td><td>path to a file which will contain trimming statistics</td><td>0</td><td>0</td></tr> 						<tr><td>shortenReads</td><td>integer</td><td></td><td>0</td><td>*</td><td>shorten reads to a maximal length after trimming; positive values keep the beginning of reads; negative ones the ends (starting from cutadapt version 1.17)</td><td>0</td><td>0</td></tr> 						<tr><td>cutFixedLength</td><td>integer</td><td>[-1000000, 1000000]</td><td>0</td><td>*</td><td>trimmes a fixed length from the beginning (positive numbers) or the end of the reads (negative numbers)</td><td>0</td><td>0</td></tr> 						<tr><td>qualityCutoff</td><td>double</td><td></td><td>0</td><td>*</td><td>trimmes reads at the ends using a sliding window approach</td><td>0</td><td>0</td></tr> 						<tr><td>qualityBase</td><td>integer</td><td></td><td>33</td><td>*</td><td>base quality value</td><td>0</td><td>0</td></tr> 						<tr><td>noIndels</td><td>boolean</td><td></td><td>false</td><td>*</td><td>does not allow indels between read and adapter</td><td>0</td><td>0</td></tr> 						<tr><td>discardTrimmed</td><td>boolean</td><td></td><td>false</td><td>*</td><td>discard sequences which were trimmed</td><td>0</td><td>0</td></tr> 						<tr><td>discardUntrimmed</td><td>boolean</td><td></td><td>false</td><td>*</td><td>discard sequences which were not trimmed</td><td>0</td><td>0</td></tr> 						<tr><td>maskAdapters</td><td>boolean</td><td></td><td>false</td><td>*</td><td>does not cut the adapters but replace the corresponding regions with N</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable3" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>cutadaptTrimFile</td><td>string</td><td>absolute path to the trimmed output file</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>cutadaptInfoFile</td><td>string</td><td>absolute path to a file containing statistical values</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Cutadapt (%SOFTWARE_VERSION%) was used to remove adapters and trim sequences [Martin, Marcel. "Cutadapt removes adapter sequences from high-throughput sequencing reads." EMBnet.journal [Online], 17.1 (2011): pp. 10-12. Web. 14 Mar. 2019].</p>
					<p>Pubmed references:

						
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://github.com/marcelm/cutadapt" data-uk-tooltip="title: Website" target="_blank">https://github.com/marcelm/cutadapt</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat5-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§FILE UTILS§" meta-update="2019-02-11" meta-search="true" meta-id="52">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">File Utils</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full52" onclick="initFullModule(52, 1)" uk-toggle>deleteFolder</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">deletes a folder and all its content.</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full52" class="uk-modal-full" meta-id="52" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>deleteFolder</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch52">deletes a folder and all its content.</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">Python3</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable52" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>folder</td><td>folder</td><td>path to existing folder</td><td></td><td>1</td><td>path to the folder that will be deleted</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable52" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat2-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§GENERAL§" meta-update="2018-10-31" meta-search="true" meta-id="11">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">General</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2018-10-31
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full11" onclick="initFullModule(11, 1)" uk-toggle>env</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">prints the currently set environment variables to the standard output stream</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://www.gnu.org/software/coreutils/env/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full11" class="uk-modal-full" meta-id="11" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>env</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch11">prints the currently set environment variables to the standard output stream</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU env</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable11" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@PARAMETER@}
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable11" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.gnu.org/software/coreutils/env/" data-uk-tooltip="title: Website" target="_blank">https://www.gnu.org/software/coreutils/env/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§;§QUALITY CONTROL§" meta-update="2018-10-31" meta-search="true" meta-id="14">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2018-10-31
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full14" onclick="initFullModule(14, 1)" uk-toggle>fastQC</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">generates quality reports for sequencing data using fastQC</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full14" class="uk-modal-full" meta-id="14" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>fastQC</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch14">generates quality reports for sequencing data using fastQC</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">fastQC (tested with 0.11.3)</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable14" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>contaminants</td><td>file path</td><td>absolute</td><td></td><td>*</td><td>absolute path to a file containing non-default contaminants to screen for overrepresented sequences; format: name[TAB]sequence</td><td>0</td><td>0</td></tr> 						<tr><td>adapters</td><td>file path</td><td>absolute</td><td></td><td>*</td><td>absolute path to a file containing non-default adapters to screen against the library; format: name[TAB]sequence</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td>[1,128]</td><td>1</td><td>*</td><td>number of threads to use; each will consume about 256 megabyte of memory</td><td>0</td><td>0</td></tr> 						<tr><td>fastq</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>absolute path to fastq file which should be analyzed</td><td>0</td><td>0</td></tr> 						<tr><td>limits</td><td>file path</td><td>absolute</td><td></td><td>*</td><td>absolute path to a file containing non-default limits for warnings/errors; must be in the same format as the limits.txt shipped with fastQC</td><td>0</td><td>0</td></tr> 						<tr><td>outdir</td><td>folder path</td><td>absolute</td><td></td><td>1</td><td>absolute path to output folder</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable14" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Quality of the sequencing data was checked using FastQC (%SOFTWARE_VERSION%) [Andrews S. (2010). FastQC: a quality control tool for high throughput sequence data. Available online at: http://www.bioinformatics.babraham.ac.uk/projects/fastqc].</p>
					<p>Pubmed references:

						
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" data-uk-tooltip="title: Website" target="_blank">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§CAROLINE FRIEDEL§" meta-category="§SEQUENCING§" meta-update="2019-10-16" meta-search="true" meta-id="43">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-10-16
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full43" onclick="initFullModule(43, 1)" uk-toggle>fastqDump</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">Downloads fastq files from the NCBI Sequence Read Archive (SRA) using the SRA toolkit. First performs prefetch and then fastq-dump. Can optionally use Aspera client ascp for much faster download (Aspera client should be installed).</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Caroline Friedel</div>
				<div class="uk-width-auto">
					<a href="https://www.ncbi.nlm.nih.gov/sra/docs/toolkitsoft/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/21062823" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full43" class="uk-modal-full" meta-id="43" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>fastqDump</h1>
					<h3><i>by</i> Caroline Friedel - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch43">Downloads fastq files from the NCBI Sequence Read Archive (SRA) using the SRA toolkit. First performs prefetch and then fastq-dump. Can optionally use Aspera client ascp for much faster download (Aspera client should be installed).</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">SRA toolkit</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable43" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>sraId</td><td>string</td><td></td><td></td><td>1</td><td>SRA id</td><td>0</td><td>0</td></tr> 						<tr><td>outputFolder</td><td>file</td><td></td><td></td><td>1</td><td>folder to which fastq files should be extracted</td><td>0</td><td>0</td></tr> 						<tr><td>pathToAspera</td><td>file</td><td></td><td></td><td>*</td><td>[optional] path to Aspera client to use Aspera to speedup download</td><td>0</td><td>0</td></tr> 						<tr><td>checkPresent</td><td>boolean</td><td></td><td>false</td><td>*</td><td>[optional] check if files already present in output folder and download previously successful. Tests if output fastq files exist, the log file from a previous download is present, fastq files are created not later than the lof file and the log files shows a succesful download. </td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable43" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>isPairedEnd</td><td>boolean</td><td>Indicates whether paired-end (two fastq files) or single-end (one fastq file) sequencing data was downloaded</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>readFile1</td><td>string</td><td>path to first fastq file</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>readFile2</td><td>string</td><td>path to second fastq file (identical to first fastq file for single-end sequencing data)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Sequencing data was downloaded from the NCBI Sequence Read Archive (SRA) using the SRA toolkit (version (%SOFTWARE_VERSION%)) [Leinonen R, Sugawara H, Shumway M; International Nucleotide Sequence Database Collaboration. The sequence read archive. Nucleic Acids Res. 2011 Jan;39(Database issue):D19-21.]</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/21062823" data-uk-tooltip="title: PMID" target="_blank">21062823</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.ncbi.nlm.nih.gov/sra/docs/toolkitsoft/" data-uk-tooltip="title: Website" target="_blank">https://www.ncbi.nlm.nih.gov/sra/docs/toolkitsoft/</a><br /> 						<a href="https://asperasoft.com" data-uk-tooltip="title: Website" target="_blank">https://asperasoft.com</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§;§GENE EXPRESSION§" meta-update="2019-03-13" meta-search="true" meta-id="29">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-03-13
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full29" onclick="initFullModule(29, 2)" uk-toggle>featureCounts</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">reads or fragments per gene, exon or any other feature are counted using featureCounts</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://subread.sourceforge.net" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/24227677" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 2
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full29" class="uk-modal-full" meta-id="29" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>featureCounts</h1>
					<h3><i>by</i> Michael Kluge - version 2
					<div class="uk-inline mayhide" meta-hide="false">
				    <div class="uk-text-small">version

														<a class="uk-link-reset uk-badge" style="margin-left: 10px;" href="#" onclick="setVersion(29, 1)">1</a> 							<a class="uk-link-reset uk-badge" style="margin-left: 10px;" href="#" onclick="setVersion(29, 2)">2</a> 
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch29">reads or fragments per gene, exon or any other feature are counted using featureCounts</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">featureCounts (v. 1.4.6)</li> 						<li class="versionMetaFilter" meta-minVersion="2" meta-maxVersion="2">featureCounts (v. 1.6.1)</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="2">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable29" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>annotation</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>feature annotation in GTF or SAF format</td><td>0</td><td>0</td></tr> 						<tr><td>input</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>indexed BAM file which should be used for counting</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to output file</td><td>0</td><td>0</td></tr> 						<tr><td>annotationType</td><td>enum</td><td>SAF|GTF</td><td></td><td>*</td><td> disables automatic type detection based on the file ending of the input file; valid values: GTF or SAF;</td><td>0</td><td>0</td></tr> 						<tr><td>featureType</td><td>string</td><td></td><td>exon</td><td>*</td><td>feature type (e.g. exon or intron) which is used for counting in GTF mode</td><td>0</td><td>0</td></tr> 						<tr><td>groupType</td><td>string</td><td></td><td>gene_id</td><td>*</td><td>attribute which is used for summarization in GTF mode</td><td>0</td><td>0</td></tr> 						<tr><td>stranded</td><td>integer</td><td></td><td>0</td><td>*</td><td>indicates strand-specific read counting; possible values: 0 (unstranded), 1 (stranded) and 2 (reversely stranded)</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td></td><td>1</td><td>*</td><td>number of threads used for counting</td><td>0</td><td>0</td></tr> 						<tr><td>disableGroupSummarization</td><td>boolean</td><td></td><td>false</td><td>*</td><td>flag that can be used to turn summarization on groupType off</td><td>0</td><td>0</td></tr> 						<tr><td>multiMapping</td><td>boolean</td><td></td><td>false</td><td>*</td><td>flag that enables counting of multi mapped reads</td><td>0</td><td>0</td></tr> 						<tr><td>primary</td><td>boolean</td><td></td><td>true</td><td>*</td><td>when enabled only alignments which are flagged as primary alignments are counted</td><td>0</td><td>0</td></tr> 						<tr><td>countFragments</td><td>boolean</td><td></td><td>false</td><td>*</td><td>counts fragments instead of reads; only for paired end data</td><td>0</td><td>0</td></tr> 						<tr><td>multiCountMetaFeatures</td><td>boolean</td><td></td><td>false</td><td>*</td><td>allows a read to be counted for more than one meta-feature</td><td>0</td><td>0</td></tr> 						<tr><td>detailedReadAssignments</td><td>boolean</td><td></td><td>false</td><td>*</td><td>saves for each read if it was assigned or not; filename: {input_file_name}.featureCounts; format: read name&lt;TAB&gt;status&lt;TAB&gt;feature name&lt;TAB&gt;number of counts for that read</td><td>0</td><td>0</td></tr> 						<tr><td>minOverlap</td><td>integer</td><td></td><td>1</td><td>*</td><td>minimum number of overlapping bases required to assign a read to a feature; also negative values are allowed</td><td>2</td><td>2</td></tr> 						<tr><td>minReadOverlap</td><td>integer</td><td></td><td>1</td><td>*</td><td>minimum number of overlapping bases required to assign a read to a feature; also negative values are allowed</td><td>1</td><td>1</td></tr> 						<tr><td>minFracOverlap</td><td>double</td><td></td><td>0</td><td>*</td><td>assign reads to the meta-feature/feature which has the largest number of overlapping bases</td><td>2</td><td>2</td></tr> 						<tr><td>readExtension5</td><td>integer</td><td></td><td>0</td><td>*</td><td>extend reads at the 5' end</td><td>2</td><td>2</td></tr> 						<tr><td>readExtension3</td><td>integer</td><td></td><td>0</td><td>*</td><td>extend reads at the 3' end</td><td>2</td><td>2</td></tr> 						<tr><td>fraction</td><td>boolean</td><td></td><td>false</td><td>*</td><td>count fractional; only in combination with the --assignToAllOverlappingFeatures or/and --multiMapping flag(s)</td><td>2</td><td>2</td></tr> 						<tr><td>largestOverlap</td><td>boolean</td><td></td><td>false</td><td>*</td><td>assign reads to the meta-feature/feature that has the largest number of overlapping bases.</td><td>2</td><td>2</td></tr> 						<tr><td>longReads</td><td>boolean</td><td></td><td>false</td><td>*</td><td>mode for long read counting (e.g. Nanopore or PacBio)</td><td>2</td><td>2</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable29" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>FeatureCountSummaryFile</td><td>string</td><td>absolute file path to the summary file</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>FeatureCountCountFile</td><td>string</td><td>absolute file path to the count file</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>FeatureCounts (%SOFTWARE_VERSION%) was applied to count read/fragment counts per gene/exon/other feature according to %annotation§N% annotation [Liao Y, Smyth GK and Shi W. featureCounts: an efficient general-purpose program for assigning sequence reads to genomic features. Bioinformatics, 30(7):923-30, 2014].</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/24227677" data-uk-tooltip="title: PMID" target="_blank">24227677</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://subread.sourceforge.net" data-uk-tooltip="title: Website" target="_blank">https://subread.sourceforge.net</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat3-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§CHIP-SEQ§" meta-update="2019-02-11" meta-search="true" meta-id="55">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">ChIP-seq</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full55" onclick="initFullModule(55, 1)" uk-toggle>filterBwaSampe</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">removes read pairs from sam/bam files created by bwa sampe</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full55" class="uk-modal-full" meta-id="55" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>filterBwaSampe</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch55">removes read pairs from sam/bam files created by bwa sampe</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">pysam package</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable55" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>inSamBam</td><td>file</td><td>file exsists, ending .sam or .bam</td><td></td><td>1</td><td>path to mapped paired reads in sam or bam format (recognized by file ending) created by bwa sampe</td><td>0</td><td>0</td></tr> 						<tr><td>outSamBam</td><td>string</td><td></td><td></td><td>1</td><td>path to write remaining paired reads in sam or bam format (recognized by file ending)</td><td>0</td><td>0</td></tr> 						<tr><td>removeUnmapped</td><td>boolean</td><td></td><td>True</td><td>*</td><td>use this flag to remove pairs with at least one unmapped read</td><td>0</td><td>0</td></tr> 						<tr><td>removeImproperPairs</td><td>boolean</td><td></td><td>True</td><td>*</td><td>use this flag to remove pairs that are not properly paired according to bwa sampe</td><td>0</td><td>0</td></tr> 						<tr><td>removeMapqBelow</td><td>integer</td><td>&gt;=0</td><td>20</td><td>*</td><td>remove all read pairs with at least one mate of mapping quality smaller than minQuality (taken from field &quot;MAPQ&quot; in SAM file), setting the option to 0 deactivates filtering based on mapping quality</td><td>0</td><td>0</td></tr> 						<tr><td>removeMoreThanOptimalHits</td><td>integer</td><td>&gt;=0</td><td>1</td><td>*</td><td>remove all read pairs with more than maxHits optimal alignment positions for at least one mate (based bwa aln specific tag &quot;X0&quot;), setting the option to 0 deactivates filtering based on hit number</td><td>0</td><td>0</td></tr> 						<tr><td>isSingleEnd</td><td>boolean</td><td></td><td>False</td><td>*</td><td>use this flag to indicate that single end data should be filtered</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable55" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>filteredPairs</td><td>file</td><td>path of the sam/bam file with the remaining read pairs (same value as given in parameter outSamBam)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>We removed read pairs with unmapped reads/ improper pair classification/ low mapping quality/ multi-mappings (adjust to options used)</p>
					<p>Pubmed references:

						
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§SEQUENCING§" meta-update="2019-02-11" meta-search="true" meta-id="0">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full0" onclick="initFullModule(0, 1)" uk-toggle>genomeCoverage</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">converts bam format to bedgraph and tdf format</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
					<a href="https://bedtools.readthedocs.io/en/latest/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/20110278" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full0" class="uk-modal-full" meta-id="0" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>genomeCoverage</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch0">converts bam format to bedgraph and tdf format</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">bedtools</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">igvtools</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable0" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>bam</td><td>file</td><td>file exists, BAM format, reads sorted by coordinates</td><td></td><td>1</td><td>path to bam file whose genome coverage should be analyzed</td><td>0</td><td>0</td></tr> 						<tr><td>genome</td><td>file</td><td>file exists, ending *.genome for an IGV genome file or ending *.chrom.sizes for a simple text file with genome sizes</td><td></td><td>*</td><td>genome file or file with chromosome sizes for the genome that was used to create the bam file, the file is required only if the tdf option is set</td><td>0</td><td>0</td></tr> 						<tr><td>outPrefix</td><td>string</td><td></td><td></td><td>1</td><td>file name prefix for saving the bedgraph file (outPrefix.bedgraph) and the tdf file (outPrefix.bedgraph.tdf)</td><td>0</td><td>0</td></tr> 						<tr><td>tdf</td><td>boolean</td><td></td><td>true</td><td>*</td><td>transform bedgraph file into tdf format using igvtools</td><td>0</td><td>0</td></tr> 						<tr><td>bedtoolsPath</td><td>file</td><td>existing executable</td><td>bedtools (in PATH)</td><td>*</td><td>path to bedtools executable, use if bedtools is not in PATH</td><td>0</td><td>0</td></tr> 						<tr><td>igvtoolsPath</td><td>file</td><td>existing executable</td><td>igvtools (in PATH)</td><td>*</td><td>path to igvtools executable, use if igvtools is not in PATH</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable0" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>We created files for visualizing mapped reads with bedtools and igvtools.</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/20110278" data-uk-tooltip="title: PMID" target="_blank">20110278</a>, 						<a href="https://www.ncbi.nlm.nih.gov/pubmed/21221095" data-uk-tooltip="title: PMID" target="_blank">21221095</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://bedtools.readthedocs.io/en/latest/" data-uk-tooltip="title: Website" target="_blank">https://bedtools.readthedocs.io/en/latest/</a><br /> 						<a href="https://software.broadinstitute.org/software/igv/igvtools_commandline" data-uk-tooltip="title: Website" target="_blank">https://software.broadinstitute.org/software/igv/igvtools_commandline</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat2-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§GENERAL§" meta-update="2018-10-31" meta-search="true" meta-id="24">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">General</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2018-10-31
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full24" onclick="initFullModule(24, 1)" uk-toggle>grep</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">extracts information from text files using the exact or regex-based search of grep</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://www.gnu.org/software/grep/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full24" class="uk-modal-full" meta-id="24" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>grep</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch24">extracts information from text files using the exact or regex-based search of grep</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU grep</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable24" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>outputFile</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>absolute path to file in which the output of grep is written</td><td>0</td><td>0</td></tr> 						<tr><td>file</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>absolute path to file to use as search input</td><td>0</td><td>0</td></tr> 						<tr><td>options</td><td>string</td><td></td><td></td><td>*</td><td>additional flags or parameters that are directly delivered to grep</td><td>0</td><td>0</td></tr> 						<tr><td>pattern</td><td>string</td><td></td><td></td><td>1</td><td>pattern to search for; can also be a regex if parameter -P is set</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable24" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>grepResultFile</td><td>string</td><td>path to the output file</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.gnu.org/software/grep/" data-uk-tooltip="title: Website" target="_blank">https://www.gnu.org/software/grep/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat{@CAT_ID@}-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§RNA-SEQ§" meta-update="2019-02-11" meta-search="true" meta-id="33">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">RNA-seq</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full33" onclick="initFullModule(33, 1)" uk-toggle>gseaPreranked</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">peforms gene set enrichment analysis with GSEAPreranked</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
					<a href="http://software.broadinstitute.org/gsea/index.jsp" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/16199517" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full33" class="uk-modal-full" meta-id="33" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>gseaPreranked</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch33">peforms gene set enrichment analysis with GSEAPreranked</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">java8</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GSEA</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable33" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>gseaJar</td><td>file</td><td></td><td></td><td>1</td><td>Path of the GSEA jar file</td><td>0</td><td>0</td></tr> 						<tr><td>label</td><td>string</td><td></td><td></td><td>1</td><td>name of the analysis, e.g. sample name</td><td>0</td><td>0</td></tr> 						<tr><td>outdir</td><td>string</td><td></td><td></td><td>1</td><td>directory to store the results of GSEA</td><td>0</td><td>0</td></tr> 						<tr><td>geneTab</td><td>file</td><td>file exists</td><td></td><td>1</td><td>tab-separated table of genes with expression values/changes</td><td>0</td><td>0</td></tr> 						<tr><td>hasHeader</td><td>boolean</td><td></td><td>False</td><td>*</td><td>indicates if the first line of the geneTab should be interpreted as header</td><td>0</td><td>0</td></tr> 						<tr><td>geneCol</td><td>integer</td><td>&gt;=0</td><td>0</td><td>*</td><td>0-based position of the column with gene names</td><td>0</td><td>0</td></tr> 						<tr><td>rankCol</td><td>integer</td><td>&gt;=0</td><td>1</td><td>*</td><td>0-based position of the column with values to rank the genes, e.g. fold changes</td><td>0</td><td>0</td></tr> 						<tr><td>geneset</td><td>string</td><td>allowed values: go, hallmark, transcription_factor, oncogenic_signatures, immunologic_signatures</td><td>hallmark</td><td>*</td><td>gene sets to test for enrichment</td><td>0</td><td>0</td></tr> 						<tr><td>genesetVersion</td><td>string</td><td></td><td>6.1</td><td>*</td><td>version of MSigDB to use</td><td>0</td><td>0</td></tr> 						<tr><td>scoring</td><td>string</td><td>allowed values: weighted, unweighted</td><td>unweighted</td><td>*</td><td>unweighted: classic score based on ranks, weighted: score includes values used for ranking</td><td>0</td><td>0</td></tr> 						<tr><td>plotNr</td><td>integer</td><td>gt;0</td><td>50</td><td>*</td><td>create plots for &quot;plot_nr&quot; top scoring genes</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable33" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>We performed gene set enrichment analysis with GSEAPreranked.</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/16199517" data-uk-tooltip="title: PMID" target="_blank">16199517</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="http://software.broadinstitute.org/gsea/index.jsp" data-uk-tooltip="title: Website" target="_blank">http://software.broadinstitute.org/gsea/index.jsp</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-08-08" meta-search="true" meta-id="44">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SEQUENCING</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-08-08
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full44" onclick="initFullModule(44, 1)" uk-toggle>gtf2info</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">extracts information on genes and exons from GTF files and stores it in CSV format</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full44" class="uk-modal-full" meta-id="44" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>gtf2info</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch44">extracts information on genes and exons from GTF files and stores it in CSV format</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">perl</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable44" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>gtf</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to the GTF file</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to the output file; for exons suffix '.exons' is added</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable44" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>geneInfoFile</td><td>string</td><td>absolute path to the resulting CSV file</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat2-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§GENERAL§" meta-update="2018-10-31" meta-search="true" meta-id="22">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">General</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2018-10-31
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full22" onclick="initFullModule(22, 1)" uk-toggle>gzip</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">compresses and decompresses files using gzip; is able to verify file integrity using a md5 checksum file</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://www.gnu.org/software/gzip/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full22" class="uk-modal-full" meta-id="22" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>gzip</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch22">compresses and decompresses files using gzip; is able to verify file integrity using a md5 checksum file</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU gzip or pigz</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable22" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>output</td><td>absolute file path</td><td></td><td>${input}.gz</td><td>*</td><td>path to output file</td><td>0</td><td>0</td></tr> 						<tr><td>input</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to input file</td><td>0</td><td>0</td></tr> 						<tr><td>decompress</td><td>boolean</td><td></td><td>false</td><td>*</td><td>decompress the input file instead of compressing it</td><td>0</td><td>0</td></tr> 						<tr><td>verify</td><td>boolean</td><td></td><td>true</td><td>*</td><td>verify file integrity after decompression using the md5 checksum file</td><td>0</td><td>0</td></tr> 						<tr><td>oldPathMd5</td><td>file path</td><td>absolute</td><td></td><td>*</td><td>path where the files was stored when the md5 checksum was created</td><td>0</td><td>0</td></tr> 						<tr><td>limitLines</td><td>integer</td><td>[1,]</td><td></td><td>*</td><td>extract only the first N lines</td><td>0</td><td>0</td></tr> 						<tr><td>delete</td><td>boolean</td><td></td><td>false</td><td>*</td><td>delete the file after compression was performed; enforces integrity check</td><td>0</td><td>0</td></tr> 						<tr><td>md5</td><td>file path</td><td>absolute</td><td></td><td>*</td><td>path to md5 checksum file to verify file integrity after decompression</td><td>0</td><td>0</td></tr> 						<tr><td>quality</td><td>integer</td><td>[1,9]</td><td>9</td><td>*</td><td>compression quality ranging from 1 to 9; 9 being the slowest but best compression</td><td>0</td><td>0</td></tr> 						<tr><td>binaryName</td><td>enum</td><td></td><td>gzip</td><td>*</td><td>name of the gzip binary; possible values: 'gzip' or 'pigz'</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td>[1,128]</td><td>1</td><td>*</td><td>number of cores to use; only possible if 'pigz' is used as binary</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable22" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>processedGzipFile</td><td>string</td><td>path to the input file</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>createdGzipFile</td><td>string</td><td>path to the output file</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.gnu.org/software/gzip/" data-uk-tooltip="title: Website" target="_blank">https://www.gnu.org/software/gzip/</a><br /> 						<a href="https://zlib.net/pigz/" data-uk-tooltip="title: Website" target="_blank">https://zlib.net/pigz/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-03-13" meta-search="true" meta-id="53">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-03-13
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full53" onclick="initFullModule(53, 1)" uk-toggle>indexBam</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">creates an index for a BAM file using samtools index</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://www.htslib.org/doc/samtools.html" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/19505943" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full53" class="uk-modal-full" meta-id="53" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>indexBam</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch53">creates an index for a BAM file using samtools index</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">samtools</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable53" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>bam</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to the BAM file</td><td>0</td><td>0</td></tr> 						<tr><td>link</td><td>boolean</td><td></td><td>true</td><td>*</td><td>creates a link called NAME.bam.bai because some tool expect the index under that name; use --nolink to disable it</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable53" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>BAMFile</td><td>string</td><td>path to the BAM file for which the index was created</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Samtools (%SOFTWARE_VERSION%) was used to index the BAM files [Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, Marth G, Abecasis G, Durbin R, and 1000 Genome Project Data Processing Subgroup, The Sequence alignment/map (SAM) format and SAMtools, Bioinformatics (2009) 25(16) 2078-9].</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/19505943" data-uk-tooltip="title: PMID" target="_blank">19505943</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.htslib.org/doc/samtools.html" data-uk-tooltip="title: Website" target="_blank">https://www.htslib.org/doc/samtools.html</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§CAROLINE FRIEDEL§" meta-category="§SEQUENCING§" meta-update="2022-03-23" meta-search="true" meta-id="2">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2022-03-23
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full2" onclick="initFullModule(2, 1)" uk-toggle>insertSizeMetrics</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">This tool provides useful metrics for validating library construction including the insert size distribution and read orientation of paired-end libraries. The expected proportions of these metrics vary depending on the type of library preparation used, resulting from technical differences between pair-end libraries and mate-pair libraries. For a brief primer on paired-end sequencing and mate-pair reads, see the GATK Dictionary. The CollectInsertSizeMetrics tool outputs the percentages of read pairs in each of the three orientations (FR, RF, and TANDEM) as a histogram. In addition, the insert size distribution is output as both a histogram (.insert_size_Histogram.pdf) and as a data table (.insert_size_metrics.txt). Note: Metrics labeled as percentages are actually expressed as fractions!</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Caroline Friedel</div>
				<div class="uk-width-auto">
					<a href="https://broadinstitute.github.io/picard/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full2" class="uk-modal-full" meta-id="2" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>insertSizeMetrics</h1>
					<h3><i>by</i> Caroline Friedel - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch2">This tool provides useful metrics for validating library construction including the insert size distribution and read orientation of paired-end libraries. The expected proportions of these metrics vary depending on the type of library preparation used, resulting from technical differences between pair-end libraries and mate-pair libraries. For a brief primer on paired-end sequencing and mate-pair reads, see the GATK Dictionary. The CollectInsertSizeMetrics tool outputs the percentages of read pairs in each of the three orientations (FR, RF, and TANDEM) as a histogram. In addition, the insert size distribution is output as both a histogram (.insert_size_Histogram.pdf) and as a data table (.insert_size_metrics.txt). Note: Metrics labeled as percentages are actually expressed as fractions!</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">picard</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">java</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable2" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>Histogram_FILE</td><td>file</td><td></td><td></td><td>1</td><td>File to write insert size Histogram chart to.  Required.</td><td>0</td><td>0</td></tr> 						<tr><td>INPUT</td><td>file</td><td></td><td></td><td>1</td><td>Input SAM/BAM/CRAM file.  Required.</td><td>0</td><td>0</td></tr> 						<tr><td>OUTPUT</td><td>file</td><td></td><td></td><td>1</td><td>The file to write the output to.  Required.</td><td>0</td><td>0</td></tr> 						<tr><td>arguments_file</td><td>file</td><td></td><td></td><td>0-null</td><td>[optional] read one or more arguments files and add them to the command line  This argument may be specified 0 or more times. Default value: null.</td><td>0</td><td>0</td></tr> 						<tr><td>COMPRESSION_LEVEL</td><td>integer</td><td></td><td></td><td>*</td><td>[optional] Compression level for all compressed files created (e.g. BAM and VCF).  Default value: 5.</td><td>0</td><td>0</td></tr> 						<tr><td>DEVIATIONS</td><td>double</td><td></td><td></td><td>*</td><td>[optional] Generate mean, sd and plots by trimming the data down to MEDIAN + DEVIATIONS*MEDIAN_ABSOLUTE_DEVIATION. This is done because insert size data typically includes enough anomalous values from chimeras and other artifacts to make the mean and sd grossly misleading regarding the real distribution.  Default value: 10.0.</td><td>0</td><td>0</td></tr> 						<tr><td>GA4GH_CLIENT_SECRETS</td><td>string</td><td></td><td></td><td>*</td><td>[optional] Google Genomics API client_secrets.json file path.  Default value: client_secrets.json.</td><td>0</td><td>0</td></tr> 						<tr><td>HISTOGRAM_WIDTHW</td><td>integer</td><td></td><td></td><td>*</td><td>null</td><td>0</td><td>0</td></tr> 						<tr><td>MAX_RECORDS_IN_RAM</td><td>integer</td><td></td><td></td><td>*</td><td>[optional] When writing files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort the file, and increases the amount of RAM needed.  Default value: 500000.</td><td>0</td><td>0</td></tr> 						<tr><td>METRIC_ACCUMULATION_LEVEL</td><td>string</td><td></td><td></td><td>*</td><td>[optional] The level(s) at which to accumulate metrics.    This argument may be specified 0 or more times. Default value: [ALL_READS]. Possible values: {ALL_READS, SAMPLE, LIBRARY, READ_GROUP}</td><td>0</td><td>0</td></tr> 						<tr><td>MIN_HISTOGRAM_WIDTH</td><td>integer</td><td></td><td></td><td>*</td><td>[optional] Minimum width of histogram plots. In the case when the histogram would otherwise betruncated to a shorter range of sizes, the MIN_HISTOGRAM_WIDTH will enforce a minimum range.  Default value: null. </td><td>0</td><td>0</td></tr> 						<tr><td>MINIMUM_PCT</td><td>double</td><td></td><td></td><td>*</td><td>[optional] When generating the Histogram, discard any data categories (out of FR, TANDEM, RF) that have fewer than this percentage of overall reads. (Range: 0 to 1).  Default value: 0.05.</td><td>0</td><td>0</td></tr> 						<tr><td>REFERENCE_SEQUENCE</td><td>file</td><td></td><td></td><td>*</td><td>[optional] Reference sequence file.  Default value: null.</td><td>0</td><td>0</td></tr> 						<tr><td>STOP_AFTER</td><td>integer</td><td></td><td></td><td>*</td><td>[optional] Stop after processing N reads, mainly for debugging.  Default value: 0.</td><td>0</td><td>0</td></tr> 						<tr><td>TMP_DIR</td><td>file</td><td></td><td></td><td>0-null</td><td>[optional] One or more directories with space available to be used by this program for temporary storage of working files  This argument may be specified 0 or more times. Default value: null.</td><td>0</td><td>0</td></tr> 						<tr><td>VALIDATION_STRINGENCY</td><td>string</td><td></td><td>STRICT</td><td>*</td><td>[optional] Validation stringency for all SAM files read by this program.  Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.  Default value: STRICT. Possible values: {STRICT, LENIENT, SILENT} </td><td>0</td><td>0</td></tr> 						<tr><td>VERBOSITY</td><td>string</td><td></td><td></td><td>*</td><td>[optional] Control verbosity of logging.  Default value: INFO. Possible values: {ERROR, WARNING, INFO, DEBUG}</td><td>0</td><td>0</td></tr> 						<tr><td>ASSUME_SORTED</td><td>boolean</td><td></td><td>true</td><td>*</td><td>[optional] If true (default), then the sort order in the header file will be ignored.  Default value: true. Possible values: {true, false}</td><td>0</td><td>0</td></tr> 						<tr><td>CREATE_INDEX</td><td>boolean</td><td></td><td>false</td><td>*</td><td>[optional] Whether to create an index when writing VCF or coordinate sorted BAM output.  Default value: false. Possible values: {true, false}</td><td>0</td><td>0</td></tr> 						<tr><td>CREATE_MD5_FILE</td><td>boolean</td><td></td><td>false</td><td>*</td><td>[optional] Whether to create an MD5 digest for any BAM or FASTQ files created. Default value:false. Possible values: {true, false}</td><td>0</td><td>0</td></tr> 						<tr><td>INCLUDE_DUPLICATES</td><td>boolean</td><td></td><td>false</td><td>*</td><td>[optional] If true, also include reads marked as duplicates in the insert size histogram.  Default value: false. Possible values: {true, false}</td><td>0</td><td>0</td></tr> 						<tr><td>QUIET</td><td>boolean</td><td></td><td>false</td><td>*</td><td>[optional] Whether to suppress job-summary info on System.err.  Default value: false. Possible values: {true, false}</td><td>0</td><td>0</td></tr> 						<tr><td>USE_JDK_DEFLATER</td><td>boolean</td><td></td><td>false</td><td>*</td><td>[optional] Use the JDK Deflater instead of the Intel Deflater for writing compressed output. Default value: false. Possible values: {true, false}</td><td>0</td><td>0</td></tr> 						<tr><td>USE_JDK_INFLATER</td><td>boolean</td><td></td><td>false</td><td>*</td><td>[optional] Use the JDK Inflater instead of the Intel Inflater for reading compressed input. Default value: false. Possible values: {true, false}</td><td>0</td><td>0</td></tr> 						<tr><td>version</td><td>boolean</td><td></td><td>false</td><td>*</td><td>[optional] display the version number for this tool</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable2" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>outputHistogramFile</td><td>string</td><td>output file containing the histogram of insert sizes</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>outputBamFile</td><td>string</td><td>txt file containing insert size metrics</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Insert size metrics were calculated with the picard library (%SOFTWARE_VERSION%).</p>
					<p>Pubmed references:

						
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://broadinstitute.github.io/picard/" data-uk-tooltip="title: Website" target="_blank">https://broadinstitute.github.io/picard/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§GENERAL§;§SEQUENCING§" meta-update="2019-03-13" meta-search="true" meta-id="28">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">General</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-03-13
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full28" onclick="initFullModule(28, 1)" uk-toggle>joinFiles</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">joins two or more files together</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://www.gnu.org/software/coreutils/manual/html_node/cat-invocation.html#cat-invocation" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full28" class="uk-modal-full" meta-id="28" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>joinFiles</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch28">joins two or more files together</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU cat</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable28" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>input</td><td>string</td><td></td><td></td><td>1-</td><td>multiple input files (or input folders) in the order in which they should be joined; in pattern mode (--pattern) folder path(s) are expected</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to output file</td><td>0</td><td>0</td></tr> 						<tr><td>convertPairedEnd</td><td>boolean</td><td></td><td>false</td><td>*</td><td>special flag for joining of FASTQ files; adds /1 and /2 at the end of read names if casava format 1.8 or greater is used; default: disabled</td><td>0</td><td>0</td></tr> 						<tr><td>pattern</td><td>string</td><td></td><td></td><td>0-null</td><td>one ore more unix file pattern (e.g. *.txt) that are used to find files matching that pattern; one pattern corresponds to one input folder path; order of files to join can not be influenced</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable28" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>joinedFile</td><td>string</td><td>absolute file path to the joined file</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.gnu.org/software/coreutils/manual/html_node/cat-invocation.html#cat-invocation" data-uk-tooltip="title: Website" target="_blank">https://www.gnu.org/software/coreutils/manual/html_node/cat-invocation.html#cat-invocation</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2018-10-31" meta-search="true" meta-id="37">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2018-10-31
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full37" onclick="initFullModule(37, 1)" uk-toggle>leon</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">LEON is a reference-free method to compress high throughput sequencing data</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://gatb.inria.fr/leon-read-compression/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/26370285" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full37" class="uk-modal-full" meta-id="37" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>leon</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch37">LEON is a reference-free method to compress high throughput sequencing data</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">LEON (tested with 1.0.0)</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable37" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>input</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>absolute path to input file; supported file formats: compress: *.fastq or *.fq; decompress: *.leon.tar</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td></td><td>1</td><td>*</td><td>number of cores to use</td><td>0</td><td>0</td></tr> 						<tr><td>kmerSize</td><td>integer</td><td></td><td>31</td><td>*</td><td>k-mer size that is used for compression</td><td>0</td><td>0</td></tr> 						<tr><td>outputFolder</td><td>folder path</td><td>absolute</td><td></td><td>1</td><td>path to folder in which the compressed file is stored; resulting file will have *.leon.tar or *.fastq ending</td><td>0</td><td>0</td></tr> 						<tr><td>workingDir</td><td>folder path</td><td>absolute</td><td>/usr/local/storage/</td><td>*</td><td>path to working directory</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable37" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>createdFile</td><td>string</td><td>path to the compressed or decompressed file</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Sequencing data was (de-)compressed using LEON (%SOFTWARE_VERSION%) [G. Benoit, C. Lemaitre, D. Lavenier, E. Drezen, T. Dayris, R. Uricaru, G. Rizk. (2015) Reference-free compression of high throughput sequencing data with a probabilistic de Bruijn graph. BMC Bioinformatics, 2015, 16:288.].</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/26370285" data-uk-tooltip="title: PMID" target="_blank">26370285</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://gatb.inria.fr/leon-read-compression/" data-uk-tooltip="title: Website" target="_blank">https://gatb.inria.fr/leon-read-compression/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat2-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§GENERAL§;§FILE UTILS§" meta-update="2019-11-12" meta-search="true" meta-id="23">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">General</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-11-12
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full23" onclick="initFullModule(23, 1)" uk-toggle>listFiles</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">lists files in directories based on pattern</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full23" class="uk-modal-full" meta-id="23" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>listFiles</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch23">lists files in directories based on pattern</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU findutils</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable23" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>folder</td><td>folder</td><td>absolute</td><td></td><td>1-</td><td>one ore more input folders; one for each pattern</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>file path</td><td>absolute</td><td></td><td>*</td><td>write results to a file; one line per found file</td><td>0</td><td>0</td></tr> 						<tr><td>sep</td><td>string</td><td></td><td>,</td><td>*</td><td>separator between entries</td><td>0</td><td>0</td></tr> 						<tr><td>maxdepth</td><td>integer</td><td></td><td>0</td><td>*</td><td>descend at most n levels of folders</td><td>0</td><td>0</td></tr> 						<tr><td>pattern</td><td>string</td><td></td><td></td><td>1-</td><td>one ore more unix file pattern (e.g. *.txt) that are used to find files matching that pattern; one pattern corresponds to one input folder path</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable23" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>foundFiles</td><td>string</td><td>found files joined with the separator</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§SEQUENCING§" meta-update="2019-02-11" meta-search="true" meta-id="35">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full35" onclick="initFullModule(35, 1)" uk-toggle>mappingSummary</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">summarizes read counts remaining after different analysis steps of sequencing data</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full35" class="uk-modal-full" meta-id="35" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>mappingSummary</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch35">summarizes read counts remaining after different analysis steps of sequencing data</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">matplotlib</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">seaborn</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable35" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>basicStatsSummary</td><td>file</td><td>file exists, output of mergeStatistics module</td><td></td><td>*</td><td>Output of the Watchdog Module mergeStatistics applied on the Basic Statistics reported by FASTQC (tab-separated table, column 0: type of count, column 1: read count, column 2: file name)</td><td>0</td><td>0</td></tr> 						<tr><td>rawRegex</td><td>string</td><td>valid regular expression in python re</td><td></td><td>*</td><td>regular expression with one group expression to extract the sample name from the name of a fastq file with untrimmed reads</td><td>0</td><td>0</td></tr> 						<tr><td>trimRegex</td><td>string</td><td>valid regular expression in python re</td><td></td><td>*</td><td>regular expression with one group expression to extract the sample name from the name of a fastq file with trimmed reads</td><td>0</td><td>0</td></tr> 						<tr><td>idxstatsSummary</td><td>file</td><td>file exists, output of mergeStatistics module</td><td></td><td>*</td><td>Output of the Watchdog Module mergeStatistics applied on the Idxstatistics reported by the bamstats module (tab-separated table, column 0: chromosome, column 2: read count, column 4: file name)</td><td>0</td><td>0</td></tr> 						<tr><td>bamRegex</td><td>string</td><td>valid regular expression in python re</td><td></td><td>*</td><td>regular expression with one group expression to extract the sample name from the name of a bam file with mapped reads</td><td>0</td><td>0</td></tr> 						<tr><td>chromosomeGroupingTable</td><td>file</td><td></td><td></td><td>*</td><td>tab-separated table with a header with chromosome names in column 0 and groups in column 1</td><td>0</td><td>0</td></tr> 						<tr><td>countTable</td><td>string</td><td></td><td></td><td>1</td><td>path for writing a table with all extracted read counts</td><td>0</td><td>0</td></tr> 						<tr><td>countPlot</td><td>string</td><td></td><td></td><td>*</td><td>path for saving a summary plot of total, trimmed and mapped reads, format is identified by file ending, all formats supported by pyplot are allowed</td><td>0</td><td>0</td></tr> 						<tr><td>groupPlot</td><td>string</td><td></td><td></td><td>*</td><td>path for saving a summary plot of the fraction of mapped reads for given groups of chromosomes, format is identified by file ending, all formats supported by pyplot are allowed</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable35" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>We created figures summarizing the number of reads in our sequencing experiments before and after adapter removal and mapping.</p>
					<p>Pubmed references:

						
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§CAROLINE FRIEDEL§" meta-category="§SEQUENCING§" meta-update="2019-10-15" meta-search="true" meta-id="10">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-10-15
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full10" onclick="initFullModule(10, 1)" uk-toggle>mergeBam</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">merges 2 or more bam files using samtools</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Caroline Friedel</div>
				<div class="uk-width-auto">
					<a href="http://www.htslib.org/doc/samtools.html" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/19505943" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full10" class="uk-modal-full" meta-id="10" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>mergeBam</h1>
					<h3><i>by</i> Caroline Friedel - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch10">merges 2 or more bam files using samtools</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">samtools</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable10" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>infile</td><td>file</td><td></td><td></td><td>2-</td><td>input bam file(s)</td><td>0</td><td>0</td></tr> 						<tr><td>outfile</td><td>file</td><td></td><td></td><td>1</td><td>output bam file</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable10" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>mergedBamFile</td><td>string</td><td>output bam file (= value for parameter outfile)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>bam files were merged using samtools (Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, Marth G, Abecasis G, Durbin R; 1000 Genome Project Data Processing Subgroup. The Sequence Alignment/Map format and SAMtools. Bioinformatics. 2009 Aug 15;25(16):2078-9)</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/19505943" data-uk-tooltip="title: PMID" target="_blank">19505943</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="http://www.htslib.org/doc/samtools.html" data-uk-tooltip="title: Website" target="_blank">http://www.htslib.org/doc/samtools.html</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-08-08" meta-search="true" meta-id="48">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SEQUENCING</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-08-08
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full48" onclick="initFullModule(48, 1)" uk-toggle>mergeFeatureCounts</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">combines the output of multiple featureCounts runs in one CSV file</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://www.gnu.org/software/coreutils/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full48" class="uk-modal-full" meta-id="48" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>mergeFeatureCounts</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch48">combines the output of multiple featureCounts runs in one CSV file</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable48" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>searchFolder</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to the folder in which *.counts files are located</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>file</td><td>absolute</td><td></td><td>1</td><td>path to the output file</td><td>0</td><td>0</td></tr> 						<tr><td>statsFolder</td><td>file</td><td>absolute</td><td></td><td>*</td><td>path to merged statistic folder required for plotting</td><td>0</td><td>0</td></tr> 						<tr><td>featureAnnotation</td><td>file</td><td>absolute</td><td></td><td>*</td><td>annotation file which is joined with the count file</td><td>0</td><td>0</td></tr> 						<tr><td>featureAnnotationID</td><td>string</td><td></td><td>Geneid</td><td>*</td><td>name of the column with is used for joining</td><td>0</td><td>0</td></tr> 						<tr><td>featureAnnotationType</td><td>string</td><td></td><td>type</td><td>*</td><td>name of the column in the annotation file for which a distribution plot is created</td><td>0</td><td>0</td></tr> 						<tr><td>featureAnnotationExonLength</td><td>string</td><td></td><td>exon_length</td><td>*</td><td>name of the column that contains the exon length of the features</td><td>0</td><td>0</td></tr> 						<tr><td>noPlotting</td><td>boolean</td><td></td><td>false</td><td>*</td><td>disables the execution of R scripts</td><td>0</td><td>0</td></tr> 						<tr><td>prefixNames</td><td>boolean</td><td></td><td>false</td><td>*</td><td>prefixes the names of the features with continuous numbers</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable48" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>mergedCountFile</td><td>string</td><td>absolute path to the merged count file in CSV format</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.gnu.org/software/coreutils/" data-uk-tooltip="title: Website" target="_blank">https://www.gnu.org/software/coreutils/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§;§QUALITY CONTROL§" meta-update="2019-03-15" meta-search="true" meta-id="41">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-03-15
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full41" onclick="initFullModule(41, 1)" uk-toggle>mergeStatistics</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">takes a folder containing BAM statistics generated by the bamstats module and generates table-formated files</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full41" class="uk-modal-full" meta-id="41" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>mergeStatistics</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch41">takes a folder containing BAM statistics generated by the bamstats module and generates table-formated files</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">java11</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable41" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>type</td><td>string</td><td></td><td></td><td>1</td><td>type of the statistic merger that should be called; allowed values: FastQC, Star, BamstatsMerger, CutadaptMerger, FeatureCounts, FlagstatMerger</td><td>0</td><td>0</td></tr> 						<tr><td>inputDir</td><td>folder path</td><td>absolute</td><td></td><td>1</td><td>path to input folder</td><td>0</td><td>0</td></tr> 						<tr><td>outputDir</td><td>folder path</td><td>absolute</td><td></td><td>1</td><td> path to output folder</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable41" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>mergedFile</td><td>string</td><td>absolute path to the merged file</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>mergedType</td><td>string</td><td>type of the merger (parameter: type)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat3-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§CHIP-SEQ§" meta-update="2019-02-11" meta-search="true" meta-id="32">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">ChIP-seq</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full32" onclick="initFullModule(32, 1)" uk-toggle>phantomPeak</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">analyzes strand cross-correlation in mapped reads from ChIP-seq experiments</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
					<a href="https://github.com/kundajelab/phantompeakqualtools" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/22955991" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full32" class="uk-modal-full" meta-id="32" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>phantomPeak</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch32">analyzes strand cross-correlation in mapped reads from ChIP-seq experiments</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">R &gt;=3.1</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">spp (phantompeakqualtools)</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable32" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>inBam</td><td>string</td><td>valid file path, bam format, ending *.bam</td><td></td><td>1</td><td>Path to the bam file with mapped ChIP-seq reads. An index of the bam file is not required.</td><td>0</td><td>0</td></tr> 						<tr><td>outPrefix</td><td>string</td><td></td><td></td><td>1</td><td>Common prefix of all output files. The module produces 3 files: outPrefix.txt (summary file), outPrefix.pdf (cross-correlation plot) and outPrefix.Rdata (R session of the analysis).</td><td>0</td><td>0</td></tr> 						<tr><td>sppPath</td><td>string</td><td>valid file path to the script run_spp.R</td><td></td><td>1</td><td>Path to executable (R script) of phantompeakqualtools which is usually called run_spp.R</td><td>0</td><td>0</td></tr> 						<tr><td>rscriptPath</td><td>string</td><td>valid file path, executable</td><td>Rscript in PATH variable</td><td>*</td><td>Path to executable Rscript if not given in PATH variable</td><td>0</td><td>0</td></tr> 						<tr><td>tmpdir</td><td>string</td><td>path to existing folder</td><td>return value of the tempdir() function of R</td><td>*</td><td>Folder for writing temporary files. The tool copies the whole bam file to this location. All temporary files are extended with a random suffix.</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td>&gt;=1</td><td>1</td><td>*</td><td>Number of threads used for the calculations</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable32" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Phantompeakqualtools were used to perform quality control of the mapped ChIP-seq reads.</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/22955991" data-uk-tooltip="title: PMID" target="_blank">22955991</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://github.com/kundajelab/phantompeakqualtools" data-uk-tooltip="title: Website" target="_blank">https://github.com/kundajelab/phantompeakqualtools</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat{@CAT_ID@}-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§SRA MINING§" meta-update="2020-12-02" meta-search="true" meta-id="5">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SRA mining</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2020-12-02
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full5" onclick="initFullModule(5, 1)" uk-toggle>recountReadout</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">calculates readout for every sample in a project from recount.</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
					<a href="https://jhubiostatistics.shinyapps.io/recount/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/28398307" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full5" class="uk-modal-full" meta-id="5" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>recountReadout</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch5">calculates readout for every sample in a project from recount.</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">Python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">R 3.5.x</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">R packages recount and recount.bwtool</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable5" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>projectID</td><td>string</td><td>mutually exclusive with projectFile</td><td></td><td>*</td><td>project id of a sra project indexed in recount2, it is possible to pass several project ids separated by ,</td><td>0</td><td>0</td></tr> 						<tr><td>projectFile</td><td>file</td><td>file exists, mutually exclusive with projectID</td><td></td><td>*</td><td>file with one line giving project ids (file content = all allowed values for projectID)</td><td>0</td><td>0</td></tr> 						<tr><td>geneTSV</td><td>file</td><td>file exists</td><td></td><td>1</td><td>
        tab-separated file with genes, cooridantes, exonic basepairs and upstream and downstream regions
        (requires a line with column names chr, geneid, exonic_bps, upstream_start, upstream_end, downstream_start and downstream_end)
      </td><td>0</td><td>0</td></tr> 						<tr><td>outfolder</td><td>folder</td><td></td><td></td><td>1</td><td>
        folder for saving final results, creates a subfolder for the project with a table of coverage values for every sample in the project
      </td><td>0</td><td>0</td></tr> 						<tr><td>tmpfolder</td><td>folder</td><td></td><td></td><td>1</td><td>folder for saving temporary data, creates a subfolder for the project (named projectID)</td><td>0</td><td>0</td></tr> 						<tr><td>Rscript</td><td>file</td><td>executable</td><td></td><td>*</td><td>path to Rscript executable (preferentially version 5.3)</td><td>0</td><td>0</td></tr> 						<tr><td>removeTmpSampleData</td><td>boolean</td><td></td><td>true</td><td>*</td><td>if this flag is set, temporary files for samples are deleted at the end (default behaviour)</td><td>0</td><td>0</td></tr> 						<tr><td>removeTmpProjectData</td><td>boolean</td><td></td><td>true</td><td>*</td><td>if this flag is set, temporary files for projects are deleted at the end (default behaviour)</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td>&gt;=1</td><td>1</td><td>*</td><td>number of threads to use, equivalent to number of samples processed in parallel</td><td>0</td><td>0</td></tr> 						<tr><td>downloadParallel</td><td>boolean</td><td></td><td>false</td><td>*</td><td>if this flag is set, big wig files are downloaded by in parallel (default: not set)</td><td>0</td><td>0</td></tr> 						<tr><td>localRecountFolder</td><td>folder</td><td>absolute</td><td></td><td>*</td><td>folder that can contain locally processed or already downloaded recount data; structure: projectID/rse_gene.Rdata and projectID/bw/sampleID.bw</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable5" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>
        Normalized readcounts for genes, upstream regions and downstream regions were calculated from the bigwigfiles provided by the Recount2 project.
    </p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/28398307" data-uk-tooltip="title: PMID" target="_blank">28398307</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://jhubiostatistics.shinyapps.io/recount/" data-uk-tooltip="title: Website" target="_blank">https://jhubiostatistics.shinyapps.io/recount/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat4-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§CIRCRNA§" meta-update="2019-02-11" meta-search="true" meta-id="59">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">circRNA</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full59" onclick="initFullModule(59, 1)" uk-toggle>removeLinearReads</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">removes linearly mappable reads from a circRNA prediction.</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full59" class="uk-modal-full" meta-id="59" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>removeLinearReads</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch59">removes linearly mappable reads from a circRNA prediction.</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">pysam (v0.14.1)</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable59" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>mapping</td><td>file</td><td>SAM or BAM format</td><td></td><td>1</td><td>
        path to a SAM or BAM file with mapped reads from the sample for which circRNAs were predicted (file ending is used to decide if it is SAM or BAM format)
      </td><td>0</td><td>0</td></tr> 						<tr><td>circRNAPrediction</td><td>file</td><td>file exists</td><td></td><td>1</td><td>
          predicted circRNAs from the CIRI2, circRNAfinder or the circCombination module (tab-separated, 5 columns: chromosome, start, end, strand, list of reads)
      </td><td>0</td><td>0</td></tr> 						<tr><td>circOut</td><td>file</td><td></td><td></td><td>1</td><td>
        all circRNAs from the input file with at least minReads remaining circular reads after removing all linearly mappable reads from the lists circular junction reads
      </td><td>0</td><td>0</td></tr> 						<tr><td>minReads</td><td>integer</td><td>&gt;=1</td><td>2</td><td>*</td><td>Minimum number of predicted junction reads required for writing a circRNA to the outputfile, default:2</td><td>0</td><td>0</td></tr> 						<tr><td>paired</td><td>string</td><td>'yes' or 'no'</td><td>yes</td><td>1</td><td>indicates if SAM or BAM input file contains paired-end (yes) or single-end (no) data</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable59" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>filteredCircs</td><td>file</td><td>path to circRNA predictions with the filtered lists of circular reads (same as input parameter circOut)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>We filtered the predicted circular reads by removing those reads that can be mapped elsewhere in a linear way.</p>
					<p>Pubmed references:

						
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat{@CAT_ID@}-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§RNA-SEQ§" meta-update="2019-02-11" meta-search="true" meta-id="16">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">RNA-seq</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full16" onclick="initFullModule(16, 1)" uk-toggle>rrnaFilter</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">removes rrna reads from sequencing data</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full16" class="uk-modal-full" meta-id="16" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>rrnaFilter</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch16">removes rrna reads from sequencing data</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">pysam</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">bwa</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable16" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>in1</td><td>file</td><td>file exists, allowed file endings: fastq, fq, fq.gz</td><td></td><td>1</td><td>first (gzipped) fastQ file with the sequenced reads</td><td>0</td><td>0</td></tr> 						<tr><td>in2</td><td>file</td><td>file exists, allowed file endings: fastq, fq, fq.gz</td><td></td><td>*</td><td>second (gzipped) fastQ file with the sequenced reads (for paired-end data only)</td><td>0</td><td>0</td></tr> 						<tr><td>rrnaIndex</td><td>string</td><td>filename prefix for a bwa index</td><td></td><td>1</td><td>Common prefix of bwa index files for the rRNA sequence</td><td>0</td><td>0</td></tr> 						<tr><td>out1</td><td>string</td><td>file path with file ending fastq, fq, fa or fq.gz</td><td></td><td>1</td><td>file for writing non-rRNA reads from in1 in fasta or (gzipped) fastq format</td><td>0</td><td>0</td></tr> 						<tr><td>out2</td><td>string</td><td>file path with file ending fastq, fq, fa or fq.gz</td><td></td><td>*</td><td>file for writing non-rRNA reads from in2 (for paired-end data) in fasta or (gzipped) fastq format</td><td>0</td><td>0</td></tr> 						<tr><td>sam</td><td>string</td><td>file path with file ending sam</td><td></td><td>1</td><td>sam file for writing rRNA reads from in1 and in2</td><td>0</td><td>0</td></tr> 						<tr><td>workdir</td><td>folder</td><td>folder exists</td><td>os.getcwd()</td><td>*</td><td>path to directory for writing large temporary files (content is deleted at the end of execution), default: current directory</td><td>0</td><td>0</td></tr> 						<tr><td>keepTmp</td><td>boolean</td><td></td><td>False</td><td>*</td><td>option to keep temporary files</td><td>0</td><td>0</td></tr> 						<tr><td>maxEditDistance</td><td>integer</td><td>&gt;=0</td><td>infinity</td><td>*</td><td>maximum allowed edit distance for a read alignment against rRNA</td><td>0</td><td>0</td></tr> 						<tr><td>maxMismatches</td><td>integer</td><td>&gt;=0</td><td>infinity</td><td>*</td><td>maximum allowed number of mismatches for a read alignment against rRNA</td><td>0</td><td>0</td></tr> 						<tr><td>maxIndels</td><td>integer</td><td>&gt;=0</td><td>infinity</td><td>*</td><td>maximum allowed number of indels for a read alignment against rRNA</td><td>0</td><td>0</td></tr> 						<tr><td>pairFiltering</td><td>integer</td><td>1 or 2</td><td>2</td><td>*</td><td>Number of reads of a pair required to fulfil the options above (maxEditDistance, maxMismatches, maxIndels)</td><td>0</td><td>0</td></tr> 						<tr><td>bwaPath</td><td>executable</td><td></td><td>bwa</td><td>*</td><td>path to bwa executable</td><td>0</td><td>0</td></tr> 						<tr><td>seedSize</td><td>integer</td><td>&gt;=1</td><td>25</td><td>*</td><td>size of initial seed for bwa (-k option of bwa)</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td>&gt;=1</td><td>1</td><td>*</td><td>number of threads to use for bwa (-t option of bwa)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable16" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>rrnaSAMFile</td><td>string</td><td>path to rRNA reads in SAM format (same value as given by the sam parameter)</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>filteredFQ1</td><td>string</td><td>path to non rRNA reads in FASTQ format (same value as given by the out1 parameter)</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>filteredFQ2</td><td>string</td><td>path to non rRNA reads in FASTQ format (same value as given by the out2 parameter), for single-end data the value of the return variable is set to &quot;not_defined_for_single_end&quot;</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Before mapping the reads to the reference genome we removed reads originating from rRNAs</p>
					<p>Pubmed references:

						
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-03-13" meta-search="true" meta-id="50">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-03-13
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full50" onclick="initFullModule(50, 1)" uk-toggle>sam2bam</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">converts SAM files into compressed BAM format using samtools sort</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://www.htslib.org/doc/samtools.html" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/19505943" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full50" class="uk-modal-full" meta-id="50" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>sam2bam</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch50">converts SAM files into compressed BAM format using samtools sort</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">samtools</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable50" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>sam</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to SAM file that should be compressed</td><td>0</td><td>0</td></tr> 						<tr><td>bam</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to ouput BAM file</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td></td><td>1</td><td>*</td><td>number of threads to use for compression</td><td>0</td><td>0</td></tr> 						<tr><td>quality</td><td>integer</td><td>[1, 9]</td><td>9</td><td>*</td><td>compression level; 1 is the worst/fastest and 9 is the best/slowest compression</td><td>0</td><td>0</td></tr> 						<tr><td>memory</td><td>string</td><td></td><td>768M</td><td>*</td><td>maximal memory that can be used per thread; only an estimation and might be exceeded!</td><td>0</td><td>0</td></tr> 						<tr><td>tmpFolder</td><td>folder path</td><td>absolute</td><td></td><td>*</td><td>write temporary files to that folder</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable50" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>BAMFile</td><td>string</td><td>absolute path to the resulting BAM file</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Samtools (%SOFTWARE_VERSION%) was used to convert SAM to BAM files [Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, Marth G, Abecasis G, Durbin R, and 1000 Genome Project Data Processing Subgroup, The Sequence alignment/map (SAM) format and SAMtools, Bioinformatics (2009) 25(16) 2078-9].</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/19505943" data-uk-tooltip="title: PMID" target="_blank">19505943</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.htslib.org/doc/samtools.html" data-uk-tooltip="title: Website" target="_blank">https://www.htslib.org/doc/samtools.html</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§CAROLINE FRIEDEL§" meta-category="§SEQUENCING§" meta-update="2019-11-12" meta-search="true" meta-id="30">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-11-12
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full30" onclick="initFullModule(30, 1)" uk-toggle>samtoolsView</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">runs samtools view on BAM/SAM/CRAM files</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Caroline Friedel</div>
				<div class="uk-width-auto">
					<a href="http://www.htslib.org/doc/samtools.html" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/19505943" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full30" class="uk-modal-full" meta-id="30" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>samtoolsView</h1>
					<h3><i>by</i> Caroline Friedel - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch30">runs samtools view on BAM/SAM/CRAM files</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">samtools</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable30" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>bamoutput</td><td>boolean</td><td></td><td>false</td><td>*</td><td>output BAM</td><td>0</td><td>0</td></tr> 						<tr><td>cramoutput</td><td>boolean</td><td></td><td>false</td><td>*</td><td>output CRAM (requires reference sequence)</td><td>0</td><td>0</td></tr> 						<tr><td>fastCompression</td><td>boolean</td><td></td><td>false</td><td>*</td><td>use fast BAM compression (implies bamoutput)</td><td>0</td><td>0</td></tr> 						<tr><td>uncompressedBam</td><td>boolean</td><td></td><td>false</td><td>*</td><td>uncompressed BAM output (implies bamoutput)</td><td>0</td><td>0</td></tr> 						<tr><td>includeHeader</td><td>boolean</td><td></td><td>false</td><td>*</td><td>include header in SAM output</td><td>0</td><td>0</td></tr> 						<tr><td>printOnlyHeader</td><td>boolean</td><td></td><td>false</td><td>*</td><td>print SAM header only (no alignments)</td><td>0</td><td>0</td></tr> 						<tr><td>printCounts</td><td>boolean</td><td></td><td>false</td><td>*</td><td>print only the count of matching records</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>file</td><td></td><td>stdout</td><td>*</td><td>output file name</td><td>0</td><td>0</td></tr> 						<tr><td>outputReadsNotSelected</td><td>file</td><td></td><td></td><td>*</td><td>output reads not selected by filters to FILE</td><td>0</td><td>0</td></tr> 						<tr><td>referenceLengths</td><td>file</td><td></td><td></td><td>*</td><td>FILE listing reference names and lengths (see long help)</td><td>0</td><td>0</td></tr> 						<tr><td>bedfile</td><td>file</td><td></td><td></td><td>*</td><td>only include reads overlapping this BED FILE</td><td>0</td><td>0</td></tr> 						<tr><td>readgroup</td><td>string</td><td></td><td></td><td>*</td><td>only include reads in read group STR</td><td>0</td><td>0</td></tr> 						<tr><td>readgroupFile</td><td>file</td><td></td><td></td><td>*</td><td>only include reads with read group listed in FILE</td><td>0</td><td>0</td></tr> 						<tr><td>mappingquality</td><td>integer</td><td></td><td>0</td><td>*</td><td>only include reads with mapping quality at least INT</td><td>0</td><td>0</td></tr> 						<tr><td>library</td><td>string</td><td></td><td></td><td>*</td><td>only include reads in library STR</td><td>0</td><td>0</td></tr> 						<tr><td>minquerylength</td><td>integer</td><td></td><td></td><td>*</td><td>only include reads with number of CIGAR operations consuming  query sequence at least INT</td><td>0</td><td>0</td></tr> 						<tr><td>bitsset</td><td>integer</td><td></td><td>0</td><td>*</td><td>only include reads with all bits set in INT set in FLAG</td><td>0</td><td>0</td></tr> 						<tr><td>bitsnotset</td><td>integer</td><td></td><td>0</td><td>*</td><td>only include reads with none of the bits set in INT set in FLAG</td><td>0</td><td>0</td></tr> 						<tr><td>readTagToStrip</td><td>string</td><td></td><td></td><td>*</td><td>read tag to strip (repeatable)</td><td>0</td><td>0</td></tr> 						<tr><td>collapseCIGAROperation</td><td>string</td><td></td><td></td><td>*</td><td>collapse the backward CIGAR operation</td><td>0</td><td>0</td></tr> 						<tr><td>seed</td><td>double</td><td></td><td>0</td><td>*</td><td>integer part sets seed of random number generator, rest sets fraction of templates to subsample</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>string</td><td></td><td></td><td>*</td><td>number of BAM/CRAM compression threads</td><td>0</td><td>0</td></tr> 						<tr><td>printLongHelp</td><td>string</td><td></td><td></td><td>*</td><td>print long help, including note about region specification</td><td>0</td><td>0</td></tr> 						<tr><td>inputfmtoption</td><td>string</td><td></td><td></td><td>*</td><td>Specify a single input file format option in the form  of OPTION or OPTION=VALUE</td><td>0</td><td>0</td></tr> 						<tr><td>outputfmt</td><td>string</td><td></td><td></td><td>*</td><td>Specify output format (SAM, BAM, CRAM)</td><td>0</td><td>0</td></tr> 						<tr><td>outputfmtoption</td><td>string</td><td></td><td></td><td>*</td><td>Specify a single output file format option in the form  of OPTION or OPTION=VALUE</td><td>0</td><td>0</td></tr> 						<tr><td>reference</td><td>string</td><td></td><td></td><td>*</td><td>Reference sequence FASTA FILE</td><td>0</td><td>0</td></tr> 						<tr><td>inbam</td><td>file</td><td></td><td></td><td>*</td><td>input bam file</td><td>0</td><td>0</td></tr> 						<tr><td>insam</td><td>file</td><td></td><td></td><td>*</td><td>input sam file</td><td>0</td><td>0</td></tr> 						<tr><td>incram</td><td>file</td><td></td><td></td><td>*</td><td>input cram file</td><td>0</td><td>0</td></tr> 						<tr><td>region</td><td>string</td><td></td><td></td><td>*</td><td>region selected</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable30" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>outputFile</td><td>string</td><td>output file (= value for parameter output)</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Samtools was used to convert BAM/SAM/CRAM to BAM/SAM/CRAM [Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, Marth G, Abecasis G, Durbin R; 1000 Genome Project Data Processing Subgroup. The Sequence Alignment/Map format and SAMtools. Bioinformatics. 2009 Aug 15;25(16):2078-9]</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/19505943" data-uk-tooltip="title: PMID" target="_blank">19505943</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="http://www.htslib.org/doc/samtools.html" data-uk-tooltip="title: Website" target="_blank">http://www.htslib.org/doc/samtools.html</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§CAROLINE FRIEDEL§" meta-category="§SEQUENCING§" meta-update="2019-11-12" meta-search="true" meta-id="38">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-11-12
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full38" onclick="initFullModule(38, 1)" uk-toggle>sashimiPlot</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">Performs visualization of splicing events across multiple samples using ggsashimi.</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Caroline Friedel</div>
				<div class="uk-width-auto">
					<a href="https://github.com/guigolab/ggsashimi" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/30118475" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full38" class="uk-modal-full" meta-id="38" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>sashimiPlot</h1>
					<h3><i>by</i> Caroline Friedel - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch38">Performs visualization of splicing events across multiple samples using ggsashimi.</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">ggsashimi</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python (2.7 or 3)</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">samtools (&gt;=1.3)</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">R (&gt;=3.3)</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">R package ggplot2 (&gt;=2.2.1)</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">R package data.table (&gt;=1.10.4)</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">R package gridExtra (&gt;=2.2.1)</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">R package svglite (&gt;=1.2.1), when generating output images in SVG format</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable38" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>help</td><td>string</td><td></td><td></td><td>*</td><td>show this help message and exit</td><td>0</td><td>0</td></tr> 						<tr><td>bam</td><td>file</td><td></td><td></td><td>*</td><td>Individual bam file or file with a list of bam files.  In the case of a list of files the format is tsv:  1col: id for bam file, 2col: path of bam file, 3+col:  additional columns</td><td>0</td><td>0</td></tr> 						<tr><td>coordinates</td><td>string</td><td></td><td></td><td>*</td><td>Genomic region. Format: chr:start-end (1-based)</td><td>0</td><td>0</td></tr> 						<tr><td>outprefix</td><td>string</td><td></td><td>sashimi</td><td>*</td><td>Prefix for plot file name</td><td>0</td><td>0</td></tr> 						<tr><td>outstrand</td><td>string</td><td></td><td>both</td><td>*</td><td>Only for --strand other than 'NONE'. Choose which  signal strand to plot: both, plus, minus</td><td>0</td><td>0</td></tr> 						<tr><td>mincoverage</td><td>integer</td><td></td><td>1</td><td>*</td><td>Minimum number of reads supporting a junction to be  drawn</td><td>0</td><td>0</td></tr> 						<tr><td>junctionsbed</td><td>file</td><td></td><td></td><td>*</td><td>Junction BED file name</td><td>0</td><td>0</td></tr> 						<tr><td>gtf</td><td>file</td><td></td><td></td><td>*</td><td>Gtf file with annotation (only exons is enough)</td><td>0</td><td>0</td></tr> 						<tr><td>strand</td><td>string</td><td></td><td>NONE</td><td>*</td><td>Strand specificity: NONE, SENSE, ANTISENSE,  MATE1_SENSE, MATE2_SENSE</td><td>0</td><td>0</td></tr> 						<tr><td>overlay</td><td>integer</td><td></td><td></td><td>*</td><td>Index of column with overlay levels (1-based)</td><td>0</td><td>0</td></tr> 						<tr><td>aggr</td><td>string</td><td></td><td></td><td>*</td><td>Aggregate function for overlay: mean, median,  mean_j, median_j. Use mean_j | median_j to keep  density overlay but aggregate junction counts</td><td>0</td><td>0</td></tr> 						<tr><td>colorfactor</td><td>integer</td><td></td><td></td><td>*</td><td>Index of column with color levels (1-based)</td><td>0</td><td>0</td></tr> 						<tr><td>alpha</td><td>double</td><td></td><td>0.5</td><td>*</td><td>Transparency level for density histogram</td><td>0</td><td>0</td></tr> 						<tr><td>palette</td><td>file</td><td></td><td></td><td>*</td><td>Color palette file. tsv file with at least 1 column, where  the color is the first column</td><td>0</td><td>0</td></tr> 						<tr><td>labels</td><td>integer</td><td></td><td></td><td>*</td><td>Index of column with labels (1-based)</td><td>0</td><td>0</td></tr> 						<tr><td>height</td><td>double</td><td></td><td>2</td><td>*</td><td>Height of the individual signal plot in inches</td><td>0</td><td>0</td></tr> 						<tr><td>annheight</td><td>double</td><td></td><td>1.5</td><td>*</td><td>Height of annotation plot in inches</td><td>0</td><td>0</td></tr> 						<tr><td>width</td><td>double</td><td></td><td>10</td><td>*</td><td>Width of the plot in inches</td><td>0</td><td>0</td></tr> 						<tr><td>basesize</td><td>integer</td><td></td><td>14</td><td>*</td><td>Base font size of the plot in pch</td><td>0</td><td>0</td></tr> 						<tr><td>outformat</td><td>string</td><td></td><td>pdf</td><td>*</td><td>Output file format: pdf, svg, png, jpeg, tiff</td><td>0</td><td>0</td></tr> 						<tr><td>outresolution</td><td>integer</td><td></td><td>300</td><td>*</td><td>Output file resolution in PPI (pixels per inch).  Applies only to raster output formats</td><td>0</td><td>0</td></tr> 						<tr><td>shrink</td><td>boolean</td><td></td><td>false</td><td>*</td><td>Shrink the junctions by a factor for nicer display</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable38" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Sashimi plots were created using ggsashimi [Garrido-Martín D, Palumbo E, Guigó R, Breschi A. ggsashimi: Sashimi plot revised for browser- and annotation-independent splicing visualization. PLoS Comput Biol. 2018 Aug 17,14(8):e1006360. ]</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/30118475" data-uk-tooltip="title: PMID" target="_blank">30118475</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://github.com/guigolab/ggsashimi" data-uk-tooltip="title: Website" target="_blank">https://github.com/guigolab/ggsashimi</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2020-11-03" meta-search="true" meta-id="8">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">SEQUENCING</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2020-11-03
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full8" onclick="initFullModule(8, 1)" uk-toggle>spring</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">SPRING is a reference-free method to compress high throughput sequencing data</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://github.com/shubhamchandak94/Spring" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/30535063" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full8" class="uk-modal-full" meta-id="8" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>spring</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch8">SPRING is a reference-free method to compress high throughput sequencing data</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">SPRING (tested with 1.0v1.0)</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU core utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable8" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>fastq</td><td>file</td><td></td><td></td><td>*</td><td>path to one or two (PE datasets) fastq files; possible endings: *.fastq, *.fq, *.fastq.gz or *.fq.gz file</td><td>0</td><td>0</td></tr> 						<tr><td>spring</td><td>file</td><td></td><td></td><td>1</td><td>path to compressed spring file; possible endings: *.spring or *.tar</td><td>0</td><td>0</td></tr> 						<tr><td>compress</td><td>boolean</td><td></td><td>true</td><td>*</td><td>if true the fastq files are compressed; otherwise the spring file is decompressed</td><td>0</td><td>0</td></tr> 						<tr><td>preserveOrder</td><td>boolean</td><td></td><td>true</td><td>*</td><td>preserve read order</td><td>0</td><td>0</td></tr> 						<tr><td>quality</td><td>boolean</td><td></td><td>true</td><td>*</td><td>retain quality values during compression</td><td>0</td><td>0</td></tr> 						<tr><td>ids</td><td>boolean</td><td></td><td>true</td><td>*</td><td>retain read identifiers during compression</td><td>0</td><td>0</td></tr> 						<tr><td>qualityMode</td><td>enum</td><td></td><td>lossless</td><td>*</td><td>possible values: 'lossless', 'qvz qv_ratio', 'ill_bin' or 'binary thr high low'</td><td>0</td><td>0</td></tr> 						<tr><td>long</td><td>boolean</td><td></td><td>false</td><td>*</td><td>use for compression of arbitrarily long reads</td><td>0</td><td>0</td></tr> 						<tr><td>decompressRange</td><td>string</td><td></td><td></td><td>*</td><td>decompress only reads (or read pairs for PE datasets) from start to end (both inclusive); e.g. '1 100'</td><td>0</td><td>0</td></tr> 						<tr><td>workingDir</td><td>file</td><td></td><td>/usr/local/storage/</td><td>*</td><td>path to working directory</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td></td><td>1</td><td>*</td><td>number of cores to use</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable8" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>createdFile</td><td>string</td><td>path to the compressed or decompressed file (separated by ',' in case of PE datasets)</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>isPairedEnd</td><td>boolean</td><td>true if paired-end data was processed</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>The FASTQ files were compressed using SPRING.</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/30535063" data-uk-tooltip="title: PMID" target="_blank">30535063</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://github.com/shubhamchandak94/Spring" data-uk-tooltip="title: Website" target="_blank">https://github.com/shubhamchandak94/Spring</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§" meta-update="2019-03-13" meta-search="true" meta-id="12">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-03-13
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full12" onclick="initFullModule(12, 1)" uk-toggle>sraDump</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">downloads and extracts FASTQ files from the Sequence Read Archive (SRA)</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&f=fastq-dump" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/21062823" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full12" class="uk-modal-full" meta-id="12" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>sraDump</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch12">downloads and extracts FASTQ files from the Sequence Read Archive (SRA)</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">fastq-dump or fasterq-dump</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable12" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>sraFile</td><td>file path</td><td>absolute</td><td></td><td>0-null</td><td>path to the *.sra file(s); can not be used in combination with --sraID</td><td>0</td><td>0</td></tr> 						<tr><td>sraID</td><td>string</td><td></td><td></td><td>0-null</td><td>one or more SRA ID(s); can not be used in combination with --sraFile</td><td>0</td><td>0</td></tr> 						<tr><td>rename</td><td>string</td><td></td><td></td><td>*</td><td>new basename for the resulting fastq files;</td><td>0</td><td>0</td></tr> 						<tr><td>outputFolder</td><td>folder path</td><td>absolute</td><td></td><td>1</td><td>path to folder in which the files should be extracted</td><td>0</td><td>0</td></tr> 						<tr><td>tmpFolder</td><td>folder path</td><td>absolute</td><td>/usr/local/storage</td><td>*</td><td>tmp folder; default: /usr/local/storage</td><td>0</td><td>0</td></tr> 						<tr><td>deleteOnSuccess</td><td>boolean</td><td></td><td>false</td><td>*</td><td>deletes the SRA file when extraction was successfull</td><td>0</td><td>0</td></tr> 						<tr><td>disablePrefetch</td><td>boolean</td><td></td><td>false</td><td>*</td><td>disables prefetching of the sra files</td><td>0</td><td>0</td></tr> 						<tr><td>binaryName</td><td>enum</td><td></td><td>fastq-dump</td><td>*</td><td>name of the sra-toolkit binary; possible values: 'fastq-dump' or 'fasterq-dump'</td><td>0</td><td>0</td></tr> 						<tr><td>threads</td><td>integer</td><td>[1,128]</td><td>1</td><td>*</td><td>number of cores to use; only possible if 'fasterq-dump' is used as binary</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable12" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>isPairedEnd</td><td>boolean</td><td>true, if paired end data was downloaded from SRA</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>baseName</td><td>string</td><td>absolute base name path to the created files</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>createdFiles</td><td>string</td><td>absolute path to all files that were downloaded separated by ','</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>Public samples were downloaded from the SRA (accession number: TODO %sraID%) [Leinonen R, Sugawara H, Shumway M, International Nucleotide Sequence Database Collaboration. The sequence read archive. Nucleic Acids Res. 2010;39(Database issue):D19-21.].</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/21062823" data-uk-tooltip="title: PMID" target="_blank">21062823</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&f=fastq-dump" data-uk-tooltip="title: Website" target="_blank">https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&f=fastq-dump</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat4-card module-parent" style="transform: translateY(0px);" meta-author="§SOPHIE FRIEDL§" meta-category="§CIRCRNA§" meta-update="2019-02-11" meta-search="true" meta-id="1">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">circRNA</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-02-11
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full1" onclick="initFullModule(1, 1)" uk-toggle>trimmedFastqPairFilter</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">extracts paired reads from 2 fastq files.</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Sophie Friedl</div>
				<div class="uk-width-auto">
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full1" class="uk-modal-full" meta-id="1" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>trimmedFastqPairFilter</h1>
					<h3><i>by</i> Sophie Friedl - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch1">extracts paired reads from 2 fastq files.</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">python3</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable1" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>inReads1</td><td>file</td><td>file exists, fastq format (read names without read numbers as /1)</td><td></td><td>*</td><td>path to first fastq file with reads</td><td>0</td><td>0</td></tr> 						<tr><td>inReads2</td><td>file</td><td>file exists, fastq format (read names without read numbers as /2)</td><td></td><td>*</td><td>path to second fastq file with reads</td><td>0</td><td>0</td></tr> 						<tr><td>inPrefix</td><td>string</td><td>prefix1.[fastq|fq] and prefix2.[fastq|fq] exist and meet the restrictions of inReads1 and inReads2</td><td></td><td>*</td><td>reads in two fastq files: prefix1.[fastq|fq], prefix2.[fastq|fq], can be used instead of inReads1 and inReads2</td><td>0</td><td>0</td></tr> 						<tr><td>outReads1</td><td>string</td><td></td><td></td><td>*</td><td>output file for first reads of paired data</td><td>0</td><td>0</td></tr> 						<tr><td>outReads2</td><td>string</td><td></td><td></td><td>*</td><td>output file for second reads of paired data</td><td>0</td><td>0</td></tr> 						<tr><td>outSingletons</td><td>string</td><td></td><td></td><td>*</td><td>output file for singleton reads without a mate</td><td>0</td><td>0</td></tr> 						<tr><td>outPrefix</td><td>string</td><td></td><td></td><td>*</td><td>writes output to three files: prefix1.fastq, prefix2.fastq, prefixsingleton.fastq, can be used instead of outReads1, outReads2 and outSingleton</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable1" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>pairedReads1</td><td>string</td><td>output file for first reads of paired data given in the parameters via outReads1 or outPrefix</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>singletonReads</td><td>string</td><td>output file for second reads of paired data given in the parameters via outReads2 or outPrefix</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>pairedReads2</td><td>string</td><td>output file for singleton reads without a mate given in the parameters via outReads2 or outPrefix</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>We removed all reads with missing mates from the paired-end fastq files.</p>
					<p>Pubmed references:

						
					</p>
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

						{@LINK_LIST@}
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat1-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§SEQUENCING§;§GENE EXPRESSION§" meta-update="2019-03-13" meta-search="true" meta-id="21">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">Sequencing</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-03-13
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full21" onclick="initFullModule(21, 1)" uk-toggle>umiDedup</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">unique molecular identifiers (UMIs) can be used to remove PCR duplicates</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://github.com/CGATOxford/UMI-tools" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
					<a href="https://www.ncbi.nlm.nih.gov/pubmed/28100584" data-uk-tooltip="title: PMID" target="_blank" class="uk-icon-link" data-uk-icon="icon:quote-right; ratio: 1"></a><!--PMID_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full21" class="uk-modal-full" meta-id="21" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>umiDedup</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch21">unique molecular identifiers (UMIs) can be used to remove PCR duplicates</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">umi_tools</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable21" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>bamFile</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to the BAM file; UMI must be a suffix of the fastq id separated with '_'</td><td>0</td><td>0</td></tr> 						<tr><td>outputFile</td><td>file path</td><td>absolute</td><td></td><td>1</td><td>path to the de-duplicated BAM file</td><td>0</td><td>0</td></tr> 						<tr><td>deleteOnSuccess</td><td>boolean</td><td></td><td>false</td><td>*</td><td>deletes the BAM file when deduplication was successfull</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable21" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>deduplicatedFile</td><td>string</td><td>absolute path to the de-duplicated BAM file</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>UMI-tools was used to remove PCR duplicates from the raw sequecing data based on UMIs [Smith T, Heger A, Sudbery I. UMI-tools: modeling sequencing errors in Unique Molecular Identifiers to improve quantification accuracy. Genome Res. 2017;27(3):491-499.].</p>
					<p>Pubmed references:

												<a href="https://www.ncbi.nlm.nih.gov/pubmed/28100584" data-uk-tooltip="title: PMID" target="_blank">28100584</a>, 
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://github.com/CGATOxford/UMI-tools" data-uk-tooltip="title: Website" target="_blank">https://github.com/CGATOxford/UMI-tools</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat2-card module-parent" style="transform: translateY(0px);" meta-author="§CAROLINE FRIEDEL§" meta-category="§GENERAL§" meta-update="2019-10-16" meta-search="true" meta-id="49">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">General</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-10-16
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full49" onclick="initFullModule(49, 1)" uk-toggle>untar</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">extracts *.tar, *.tar.gz and *.tar.bz2 archives</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Caroline Friedel</div>
				<div class="uk-width-auto">
					<a href="https://www.gnu.org/software/tar/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full49" class="uk-modal-full" meta-id="49" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>untar</h1>
					<h3><i>by</i> Caroline Friedel - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch49">extracts *.tar, *.tar.gz and *.tar.bz2 archives</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU tar</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable49" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>infile</td><td>file</td><td></td><td></td><td>1</td><td>input file, must be *tar, *tar.gz or *tar.bz2</td><td>0</td><td>0</td></tr> 						<tr><td>outputdir</td><td>file</td><td></td><td></td><td>*</td><td>[optional] output directory for extracting archive</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable49" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

						{@RETURN_VALUES@}
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.gnu.org/software/tar/" data-uk-tooltip="title: Website" target="_blank">https://www.gnu.org/software/tar/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

<div class="cat2-card module-parent" style="transform: translateY(0px);" meta-author="§MICHAEL KLUGE§" meta-category="§GENERAL§" meta-update="2019-03-13" meta-search="true" meta-id="7">
	<!-- overview card -->
	<div class="uk-card uk-card-small uk-card-default">
		<div class="uk-card-header">
			<div class="uk-grid uk-grid-small uk-text-small" data-uk-grid>
				<div class="uk-width-expand">
					<span class="cat-txt">General</span>
				</div>
				<div class="uk-width-auto uk-text-right uk-text-muted">
					<span data-uk-icon="icon:history; ratio: 1"></span> 2019-03-13
				</div>
			</div>
		</div>
		<div class="uk-card-body">
			<h5 class="uk-margin-small-bottom uk-margin-remove-adjacent uk-text-bold"><a class="uk-link-heading" href="#module-full7" onclick="initFullModule(7, 1)" uk-toggle>wget</a></h5>
			<p class="uk-text-small uk-text-muted versionSwitchDescription">wget is used to locate and download URI resources</p>
		</div>
		<div class="uk-card-footer">
			<div class="uk-grid uk-grid-small uk-grid-divider uk-flex uk-flex-middle" data-uk-grid>
				<div class="uk-width-expand uk-text-small uk-text-nowrap">Michael Kluge</div>
				<div class="uk-width-auto">
					<a href="https://www.gnu.org/software/wget/" data-uk-tooltip="title: Website" target="_blank" class="uk-icon-link" data-uk-icon="icon:link; ratio: 1"></a><!--URL_LINK-->
				</div>
				<!--<div class="uk-width-auto uk-text-right">
					<span data-uk-icon="icon:tag; ratio: 1" data-uk-tooltip="version"></span> 1
				</div>-->
			</div>
		</div>
	</div>
	<!-- full description -->
	<div id="module-full7" class="uk-modal-full" meta-id="7" uk-modal>
		<div class="uk-modal-dialog">
			<button class="uk-modal-close-full uk-close-large" type="button" uk-close></button>
			<div class="uk-grid-collapse uk-flex-middle" uk-grid>
				<div class="uk-background-cover" uk-height-viewport></div>
				<div class="uk-padding-large uk-width-auto">
					<h1>wget</h1>
					<h3><i>by</i> Michael Kluge - version 1
					<div class="uk-inline mayhide" meta-hide="true">
				    <div class="uk-text-small">version

							{@VERSION_LINKS@}
						</div>
					</div>
					</h3>
					<div class="mayhide" meta-hide="false">
					<p id="description-switch7">wget is used to locate and download URI resources</p>
					<h3 class="uk-h3"><a href="#auto-and-expand">Dependencies</a></h3>
					<ul class="uk-list uk-list-bullet">

												<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU wget</li> 						<li class="versionMetaFilter" meta-minVersion="1" meta-maxVersion="1">GNU Core Utilities</li> 
					</ul><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Parameter</a></h3>
					<table width="100%" class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="parameterTable7" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>restrictions</th>
								<th>default</th>
								<th>occurrence</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr><td>uri</td><td>string</td><td></td><td></td><td>1-</td><td>one ore more URI(s) pointing to the resource(s) to download</td><td>0</td><td>0</td></tr> 						<tr><td>output</td><td>folder path</td><td>absolute</td><td></td><td>1</td><td>path to a folder in which the downloaded files should be stored; filename remains untouched</td><td>0</td><td>0</td></tr> 						<tr><td>rename</td><td>string</td><td></td><td></td><td>0-null</td><td>renames the file to that name; multiple names must be provided in case of multiple URIs</td><td>0</td><td>0</td></tr> 						<tr><td>disableSizeCheck</td><td>boolean</td><td></td><td>false</td><td>*</td><td>flag that can be used to disable the size check that checks if a file is greater than 1KB</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
				</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Return values</a></h3>
					<table class="uk-table uk-table-justify uk-table-striped uk-table-hover uk-width-1-1" id="returnTable7" style="width:100%">
						<thead>
							<tr>
								<th>name</th>
								<th>type</th>
								<th>description</th>
								<th>minV</th>
								<th>maxV</th>
							</tr>
						</thead>

												<tr class="minVersion-0 maxVersion-0"><td>downloadedFolder</td><td>string</td><td>path to the folder in which the files were stored</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>numberOfFiles</td><td>integer</td><td>number of files that were downloaded</td><td>0</td><td>0</td></tr> 						<tr class="minVersion-0 maxVersion-0"><td>downloadedFiles</td><td>string</td><td>absolute path to the downloaded file(s) separated by ','</td><td>0</td><td>0</td></tr> 
					</table><br /><br />
					</div>
					<div class="mayhide" meta-hide="true">
					<h3 class="uk-h3"><a href="#auto-and-expand">Citation info</a></h3>
					<p>{@PAPER_DESC@}</p>
					<p>Pubmed references:

						{@PMID_LIST@}
					</p>
					</div>
					<div class="mayhide" meta-hide="false">
					<h3 class="uk-h3"><a href="#auto-and-expand">Links</a></h3>

												<a href="https://www.gnu.org/software/wget/" data-uk-tooltip="title: Website" target="_blank">https://www.gnu.org/software/wget/</a><br /> 
					</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>

	</div>
</div>
</section>
<!--/MODULE WRAPPER-->
<!--FOOTER-->
<footer id="site-foot" class="uk-section uk-section-secondary uk-section-xsmall">
	<div class="uk-container uk-text-small uk-text-center">
		<a href="https://github.com/klugem/watchdog" title="Visit Watchdog on Github" target="_blank" data-uk-tooltip>Watchdog's module reference book</a> | Built with <a href="http://getuikit.com" title="Visit UIkit 3 site" target="_blank" data-uk-tooltip><span data-uk-icon="uikit"></span></a> and <a href="https://github.com/zzseba78/Kick-Off" title="Visit Kick-Off site" target="_blank" data-uk-tooltip>Kick-Off</a>
	</div>
</footer>
<!--/FOOTER-->


<!-- JS FILES -->
<script src="js/uikit.min.js"></script>
<script src="js/uikit-icons.min.js"></script>
<script src="js/jquery.min.js"></script>
<script src="js/meta.json"></script>
<script src="js/moduleFilter.js"></script>
<script src="https://cdn.datatables.net/1.10.19/js/jquery.dataTables.min.js"></script>
</body>
</html>
